---
title: "Analysis of diabetes data set"
author: "Selina Ernst"
date: "2 5 2021"
output: html_document
---

# libraries
```{r libraries, include=FALSE}
# Bioconductor
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)

library(rstudioapi) 
library(tidyverse) 
library(ggplot2) 
library(pheatmap)
library(stats)
library(dplyr)
library(ggfortify) #new
library(reshape2) #new
library(ggrepel) #new

```

# functions
## combineGeneExprs
```{r}
# ------------------------------------------------
# function: genes in expression matrix that are mentioned more than once are combined (is sum or median better?)
# ------------------------------------------------

combineGeneExprs_median <- function(exprsmatrix){
  # all unique genes
  genes <- unique(rownames(exprsmatrix))
  # empty matrix
  result <- matrix(ncol = ncol(exprsmatrix), nrow = 0)
  # for each gene: find indices, combine rows by using median
  for (i in genes){
    indices <- which(rownames(exprsmatrix) == i)
    gene_matrix <- exprsmatrix[indices,]
    # combine values only if gene_matrix has 2 or more rows
    if (length(indices) > 1){
      # for each microarray: compute median of gene expression for certain gene 
      combined <-  apply(gene_matrix,2,function(x){median(x)})
      result <- rbind(result, combined, deparse.level = 1)}
    # if there is only one row: no need to combine values
    else if (length(indices) == 1){result = rbind(result, gene_matrix)}
  }
  rownames(result) = genes
  # output
  result
}

combineGeneExprs_sum <- function(exprsmatrix){
  # all unique genes
  genes <- unique(rownames(exprsmatrix))
  # empty matrix
  result <- matrix(ncol = ncol(exprsmatrix), nrow = 0)
  # for each gene: find indices, combine rows by using sum
  for (i in genes){
    indices <- which(rownames(exprsmatrix) == i)
    gene_matrix <- exprsmatrix[indices,]
    # combine values only if gene_matrix has 2 or more rows
    if (length(indices) > 1){
      # for each microarray: compute sum of gene expression for certain gene 
      combined <-  apply(gene_matrix,2,function(x){sum(x)})
      result <- rbind(result, combined, deparse.level = 1)}
    # if there is only one row: no need to combine values
    else if (length(indices) == 1){result = rbind(result, gene_matrix)}
  }
  rownames(result) = genes
  # output
  result
} 

combineGeneExprs_median2 <- function(exprsmatrix){
  # create data frame with a column of all the gene symbols
  data <- data.frame(geneSymbols = rownames(exprsmatrix),
                     exprsmatrix, 
                     row.names = NULL)
  # group data frame by gene symbols and combine values of one gene by median
  combined <- aggregate(data[-1], by = list(data$geneSymbols) , FUN = median)
  # convert data frame to matrix (only expression values, no row names)
  result <- as.matrix(combined[,-1])
  # use gene symbols as row names
  rownames(result) <- combined[,1]
  # output
  result
}

```

# expression data sets
## create vsn normalized expression matrix

Diabetes type 1 dataset, GSE53454, data is downloaded from the GEO database
```{r}
# ------------------------------------------------
# steps already done in Quality Control
# ------------------------------------------------

projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)
# get CEL files
setwd(paste(projectPath, "rawdata", "GSE53454 Diabetes type 1", sep = "/"))
data_diabetes <- ReadAffy()
# change cdf
data_diabetes@cdfName <- "HGU133Plus2_Hs_ENST"
# create expression matrix
diabetes_matrix <- exprs(data_diabetes)
# store colnames (microarray)
diabetes_microarray_information <- colnames(diabetes_matrix)
# vsnrm normalization
diabetes_vsnrma <- vsnrma(data_diabetes)
# expression matrix of vsnrm normalized data set
diabetes_vsnrma_matrix <- exprs(diabetes_vsnrma)
# cut rownames at "." ("ENST00000000233.10_at"  becomes "ENST00000000233")
rownames(diabetes_vsnrma_matrix) <- str_replace(rownames(diabetes_vsnrma_matrix),"\\..*","")

```

## information about microarray samples

```{r}
# ------------------------------------------------
# create data frame with information about microarray samples
# ------------------------------------------------

diabetes_microarrays = t(as.data.frame(
  sapply(diabetes_microarray_information, function(x){
    s = str_replace(x,"\\..*","")
    str_split(s, "_")
    })
  ))[,-c(2,3)]
colnames(diabetes_microarrays) <- c("number", "treatment", "time_h")
diabetes_microarrays <- transform(diabetes_microarrays, time_h = as.integer(str_remove(diabetes_microarrays[,"time_h"],"h")))
rownames(diabetes_microarrays) <- c(1:24)
diabetes_microarrays <- as.data.frame(diabetes_microarrays)

# how does the data frame look?
diabetes_microarrays

# replace old IDs sample names
colnames(diabetes_matrix) <- diabetes_microarrays[,"number"]
colnames(diabetes_vsnrma_matrix) <- diabetes_microarrays[,"number"]
rownames(data_diabetes@phenoData) <- diabetes_microarrays[,"number"]
rownames(data_diabetes@protocolData) <- diabetes_microarrays[,"number"]

```

# Processing of data

## remove control probes
```{r}
# ------------------------------------------------
# remove control probes from normalized data set
# ------------------------------------------------
dim(diabetes_vsnrma_matrix) # before
# 95721    24

diabetes_transcript_names = rownames(diabetes_vsnrma_matrix)

diabetes_vsnrma_matrix = diabetes_vsnrma_matrix[which(!startsWith(diabetes_transcript_names, "AFFX")),] # remove control probes (AFFX) 

dim(diabetes_vsnrma_matrix) # after
# 95659    24 
```
## convert transcript IDs to gene symbols
```{r}
# ------------------------------------------------
# convert Affymetrix IDs to gene symbols
# ------------------------------------------------

setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
annotation = read.csv("ensemble.103.txt") 
# file comes from www.ensembl.org/biomart

```


```{r}
ensemble_genes = annotation[,"Gene.stable.ID"]
ensemble_transcripts  = annotation[,"Transcript.stable.ID"]
ensemble_affyID = annotation[,"AFFY.HG.U133.Plus.2.probe"]
ensemble_symbol = annotation[,"HGNC.symbol"]

# which transcripts from the microarray are not listed in the ensemble.103.txt file?
diabetes_nomatchesIDs <- rownames(diabetes_vsnrma_matrix)[!(rownames(diabetes_vsnrma_matrix) %in% ensemble_transcripts)]
length(diabetes_nomatchesIDs)
# 112  (these transcripts dont occur in the txt file)

# remove these 112 transcripts from expression data
diabetes_vsnrma_matrix <- diabetes_vsnrma_matrix[rownames(diabetes_vsnrma_matrix) %in% ensemble_transcripts,]
dim(diabetes_vsnrma_matrix)
# 95547    24


# select only those symbols that are in the diabetes data set
names(ensemble_symbol) <- ensemble_transcripts 
diabetes_symbol <- ensemble_symbol[rownames(diabetes_vsnrma_matrix)]
# symbols already in right order for diabetes_vsnrma_matrix


# name the probes according to gene symbol
rownames(diabetes_vsnrma_matrix) <- as.character(diabetes_symbol)

# ------------------------------------------------
# check for NAs or "" 
# ------------------------------------------------

sum(is.na(rownames(diabetes_vsnrma_matrix)))
# 0 are NA

sum(rownames(diabetes_vsnrma_matrix) == "")
# 1096 are ""

```

## sort gene symbols alphabetically 
```{r}
# ------------------------------------------------
# alphabetical order + deletion of all rows with "" as gene symbol (1096)
# ------------------------------------------------

diabetes_GeneExprs <- diabetes_vsnrma_matrix[order(rownames(diabetes_vsnrma_matrix)),][1097:nrow(diabetes_vsnrma_matrix),] 

dim(diabetes_GeneExprs)
# 94451    24

head(diabetes_GeneExprs)
```

```{r}
# ------------------------------------------------
# save as .csv file 
# ------------------------------------------------

# create excel table with columns: genenames, gene expression values for each mircoarray
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
write.csv2(diabetes_GeneExprs, file = "diabetes_GeneExprs.csv", sep = ";", dec = ",")

# # can be read by using read.csv2
# setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
# test = read.csv2("diabetes_GeneExprs.csv")
# sym = test[,"X"]
# diabetes_GeneExprs = as.matrix(test[,2:ncol(test)])
# rownames(diabetes_GeneExprs) <- sym 
# diabetes_GeneExprs
# remove(sym)
# remove(test)
```


## select only pancreas specific TRAs
```{r}
# ------------------------------------------------
# minimize expression matrix: only pancreas TRAs 
# ------------------------------------------------

# get the tissue specific genes from "TRA Daten"
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
pancreas_specific_genes = read.csv("pancreas_specific_genes_human.csv")
pancreas_specific_genes = sort(pancreas_specific_genes)

# create a expression matrix with only those genes
diabetes_GeneExprs_sub = diabetes_GeneExprs[which(rownames(diabetes_GeneExprs) %in% pancreas_specific_genes),]

dim(diabetes_GeneExprs_sub) 
# 1569   24 
length(unique(rownames(diabetes_GeneExprs_sub)))
# 250 (number of TRA genes in the expression matrix)
head(diabetes_GeneExprs_sub)

```

## combine genes 
```{r}
# ------------------------------------------------
# use of own function to combine values of genes that are mentioned more than once
# ------------------------------------------------

diabetes_GeneExprs_sub_combined <- combineGeneExprs_median2(diabetes_GeneExprs_sub)
dim(combineGeneExprs_median2(diabetes_GeneExprs_sub))
# 250  24

dim(combineGeneExprs_median2(diabetes_GeneExprs))
# 18062    24
```


# plots
## distributions

```{r fig.height=12}
par(mfrow = c(5,5), mai = c(0.6,0.6,0.5,0.1))

for(x in 1:24){
  hist(diabetes_GeneExprs[,x], 
       xlab = "gene expression values",
       main = paste("Histogram of", diabetes_microarrays[x,"number"],"\n", "data set: GSE53454"))
  abline(v = c(mean(diabetes_GeneExprs[,x]), median(diabetes_GeneExprs[,x])), 
         col = c("red", "blue"), lwd = 2)
}
plot.new()
legend("center", legend = c("mean", "median"), fill = c("red", "blue"), cex = 2)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "diabetes_histogram_microarrays.pdf")

```



## heatmap of TRAs
```{r}

pheatmap(t(diabetes_GeneExprs_sub), 
         labels_row = paste(diabetes_microarrays[,"time_h"],",",diabetes_microarrays[,"treatment"]),
         labels_col = rep("",nrow(diabetes_GeneExprs_sub)))

# after combining
pheatmap(t(combineGeneExprs_median2(diabetes_GeneExprs_sub)), 
         labels_row = paste(diabetes_microarrays[,"time_h"],",",diabetes_microarrays[,"treatment"]),
         labels_col = rep("",nrow(diabetes_GeneExprs_sub)))



```
## boxplot of TRAs

```{r fig.height=20}
# ------------------------------------------------
# boxplot (very unfinished, just to get an overview)
# ------------------------------------------------
par(las = 2)
boxplot(t(combineGeneExprs_median2(diabetes_GeneExprs_sub)),
         col = rainbow(250),
         main="", 
         horizontal = TRUE)
abline(v = median(apply(diabetes_vsnrma_matrix, 1,median)))

```

# control vs. treatment

## PCA & kmeans
```{r}
# ------------------------------------------------
# PCA & kmeans
# ------------------------------------------------

# create a data frame (columns = treatment, time_h, median combined pancreas specifc TRAs)
diabetes_df <- cbind(treatment = diabetes_microarrays[,"treatment"], 
            time_h = diabetes_microarrays[,"time_h"],
            data.frame(t(diabetes_GeneExprs_sub_combined)))
head(diabetes_df)

# PCA on the TRA expressions
diabetes_pca <- prcomp(diabetes_df[,-c(1,2)])

# elbow plot 
plot(x = 1:10, y = sapply(1:10, function(i){kmeans(diabetes_pca$x,i)$tot.withinss}), 
     type = "l", xlab = "number of clusters", ylab = "total within-cluster sum of squares")
# result is not clear: k = 2 or k = 4

# plot PC1 against PC2 and show the eigenvectors (loadings)
autoplot(diabetes_pca, data = diabetes_df, colour = "treatment", size = "time_h", main = "Principal Component Analysis on beta cell data set", xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1)
autoplot(diabetes_pca, data = diabetes_df, colour = "treatment", size = "time_h", loadings = TRUE, loadings.label = TRUE, loadings.label.size = 3, main = "Principal Component Analysis on beta cell data set", xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1)
# plot kmeans for k = 2 and k = 4
autoplot(kmeans(diabetes_pca$x,2), data = diabetes_pca$x, frame = TRUE, xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1, main = "kmeans clustering (k=2) on beta cell data set")  
autoplot(kmeans(diabetes_pca$x,4), data = diabetes_pca$x, frame = TRUE, xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1, main = "kmeans clustering (k=4) on beta cell data set") 
```

```{r}
# add cluster numbers to already existing data frame (rows = microarrays)
diabetes_pca_kmeans2 <- kmeans(diabetes_pca$x,2)
diabetes_pca_kmeans4 <- kmeans(diabetes_pca$x,4)
diabetes_df <- cbind(kmeans2 = diabetes_pca_kmeans2$cluster, kmeans4 = diabetes_pca_kmeans4$cluster, diabetes_df)
(diabetes_df)

```


```{r warning=FALSE}
treshhold = 0.1 

# plot eigenvector values (PC1 vs PC2)
ggplot(diabetes_pca$rotation, aes(PC1, PC2, label = rownames(diabetes_pca$rotation))) + 
  geom_rect(xmin = -treshhold, xmax = treshhold, ymin = -treshhold, ymax = treshhold,
            alpha = 0.5, color = "red") + 
  geom_point() + 
  geom_text(col = "red", size = 2) +
  coord_equal(xlim = c(-0.6,0.6), ylim = c(-0.6,0.6)) 
  

# select the genes that contribute the most to PC1 & PC2
diabetes_pca_importantGenes = rownames(filter(data.frame(diabetes_pca$rotation), PC1 > treshhold | PC1 < -treshhold | PC2 > treshhold | PC2 < -treshhold ))

# plot only the genes that strongly influence PC1 & PC2
ggplot(diabetes_pca$rotation[diabetes_pca_importantGenes,], aes(PC1, PC2, label = diabetes_pca_importantGenes)) + 
  geom_point() +   
  geom_text(col = "red", size = 2) +
  coord_equal(xlim = c(-0.6,0.6), ylim = c(-0.6,0.6))


```

## high variance
```{r}
# ------------------------------------------------
# find genes with high variance between the microarrays 
# (can be caused by different treatment or different time point)
# ------------------------------------------------

# compute variance of each gene (each gene: 24 values)
diabetes_var = sort(apply(diabetes_GeneExprs_sub_combined,1,var), decreasing = TRUE)

# are these high variance genes the same as the genes that strongly contribute to PCA?

diabetes_pca_importantGenes %in% names(diabetes_var[diabetes_var > 2*median(diabetes_var)])
# all genes from PCA have 2x higher than the median variance

names(diabetes_var)[1:20] %in% diabetes_pca_importantGenes
# all top 20 high variance gene also contribute strongly in PCA
```

```{r}
# look at gene with highest variance (CXCL11)
diabetes_CXCL11 <- data.frame(diabetes_microarrays[,c("number","treatment", "time_h")], CXCL11 = diabetes_GeneExprs_sub_combined["CXCL11",])
ggplot(diabetes_CXCL11, aes(x = treatment, y = CXCL11)) + 
  geom_boxplot()

```

## genes that differ with treatment / time

```{r}
# ------------------------------------------------
# boxplot of high variance genes (grouped by treatment / time)
# ------------------------------------------------

# melt expression values of top50 high variance genes
mel = melt(t(diabetes_GeneExprs_sub_combined[names(diabetes_var)[1:50],]))
# vector to annotate the values from mel with treatment 
tr = diabetes_microarrays[,"treatment"]
names(tr) = diabetes_microarrays[,"number"]
tr = unname(tr[as.character(mel$Var1)])
# vector to annotate the values from mel with time (only two groups, based on kmeans2)  
ti = c(1:nrow(diabetes_microarrays))
ti[diabetes_microarrays[,"time_h"] <= 36] = "1-36"
ti[diabetes_microarrays[,"time_h"] > 36] = "48-96"
names(ti) = diabetes_microarrays[,"number"]
ti = unname(ti[as.character(mel$Var1)])

# create data frame (used for ggplot)
diabetes_highvariance_GeneExprs <- cbind(treatment = tr, time = ti, mel)
head(diabetes_highvariance_GeneExprs)
remove(tr,ti,mel) # remove these variables from Environment (not needed)

# plot boxplots of top50 high variance genes (grouped by treatment)
ggplot(diabetes_highvariance_GeneExprs, aes(x = Var2, y = value, colour = treatment)) + 
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, size = 5)) +
  xlab("Genes")
# plot boxplots of top50 high variance genes (grouped by time)
ggplot(diabetes_highvariance_GeneExprs, aes(x = Var2, y = value, colour = time)) + 
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, size = 5))+
  xlab("Genes")
```

## statistical tests
```{r}
# ------------------------------------------------
# ttest (on gene expression of top 10 high variance genes) +
# two sided wilcoxon rank sum exact test 
# ------------------------------------------------
## H1 (treatment): treatment does not change gene expression
## H1 -> alpha = 5% (p < alpha)
## H1.15 -> alpha = 15% (p < alpha)

stat_tests2 <- function(df, ctrl, treat){
  
  w = apply(df, 1, function(x){
    wilcox.test(x[ctrl], x[treat])$p.value
  })
  
  t = apply(df, 1, function(x){
    t.test(x[ctrl], x[treat])$p.value
  })
  
  data.frame(wilcox_pvalue = w, 
             H1_wilcox = w < 0.05, 
             H1.15_wilcox = w < 0.15,
             ttest_pvalue = t, 
             H1_t = t < 0.05)
}


# compare the control group against the treatment (for genes with high var)
diabetes_highvariance_treatment_stat <- stat_tests(diabetes_GeneExprs_sub_combined[names(diabetes_var)[1:50],],
           diabetes_microarrays[,"treatment"] == "Control",
           diabetes_microarrays[,"treatment"] == "Cytok")
# compare the 1-36 h group against the over 48-96 h group (for genes with high var)
diabetes_highvariance_time_stat <- stat_tests(diabetes_GeneExprs_sub_combined[names(diabetes_var)[1:50],],
           diabetes_microarrays[,"time_h"] <= 36,
           diabetes_microarrays[,"time_h"] > 36)
# compare the control group against the treatment (for genes with high var)
diabetes_treatment_stat <- stat_tests(diabetes_GeneExprs_sub_combined,
           diabetes_microarrays[,"treatment"] == "Control",
           diabetes_microarrays[,"treatment"] == "Cytok")
# compare the 1-36 h group against the over 48-96 h group (for genes with high var)
diabetes_time_stat <- stat_tests(diabetes_GeneExprs_sub_combined,
           diabetes_microarrays[,"time_h"] <= 36,
           diabetes_microarrays[,"time_h"] > 36)


diabetes_highvariance_treatment_stat
diabetes_highvariance_time_stat
diabetes_treatment_stat
diabetes_time_stat


# which high variance genes have a significant p value for treatment?
filter(diabetes_highvariance_treatment_stat, H1_wilcox)
# nrow = 12
filter(diabetes_highvariance_treatment_stat, H1_t)
# nrow = 11
diabetes_highvariance_treatment_stat
# which genes have a significant p value for time?
filter(diabetes_highvariance_time_stat, H1_wilcox)
# nrow = 32
filter(diabetes_highvariance_time_stat, H1_t)
# nrow = 34

# which genes have a significant p value for both treatment & time?
diabetes_highvariance_time_stat[diabetes_highvariance_time_stat$H1_wilcox == TRUE & diabetes_highvariance_treatment_stat$H1_wilcox == TRUE,]
#  PABPC4    REG1A 
## not part of the diabetes_pca_importantGenes (no important PCA contributor)

```

## log2foldchange

```{r}
log2.fold.change <- function(control, test){
  
  # only if control have the same row numbers
  if(nrow(control) == nrow(test)){
    
    # control / treatment (values combined by median)
    ratio = sapply(1:nrow(control), function(x){
      g1_m = median(control[x,])
      g2_m = median(test[x,])
      g1_m/g2_m
    })
    # log2 on ratio => log2(control/treatment)
    log2_foldchange = log2(ratio)
    # ttest control vs treatment (expression values)
    t = sapply(1:nrow(control), function(x){
      t.test(control[x,], test[x,])$p.value
    })
    # wilcoxon rank sum test control vs treatment (expression values)
    w = sapply(1:nrow(control), function(x){
      wilcox.test(control[x,], test[x,], exact = FALSE)$p.value
    })
    
  }else{print("Something has gone rone with the dimensions of the groups (probably row numbers)")}
  
  # output
  data.frame(GeneSymbol = rownames(control), # genes
             foldchange_ratio = ratio, # without log2
             log2_foldchange = log2_foldchange,  # log2 foldchange
             ttest_raw_pvalue = t,# ttest
             ttest_log10_pvalue = -1*log10(t),
             wilcox_raw_pvalue = w, # wilcoxon rank sum test
             wilcox_log10_pvalue = -1*log10(w)
             )

}

# gene expression matrix for each subgroup (treatment, control)
diabetes_ctrl = diabetes_GeneExprs_sub_combined[,filter(diabetes_microarrays, treatment == "Control")[,"number"]]
diabetes_treat = diabetes_GeneExprs_sub_combined[,filter(diabetes_microarrays, treatment == "Cytok")[,"number"]]

log2(median(diabetes_ctrl[1,])/median(diabetes_treat[1,]))


diabetes_fc_treatment = log2.fold.change(diabetes_ctrl, diabetes_treat)
diabetes_fc_treatment

ggplot(data = diabetes_fc_treatment, aes(x = log2_foldchange, y = ttest_log10_pvalue, label = GeneSymbol)) + 
  geom_point() + 
  geom_label_repel(size = 3, col  = "red")

diabetes_fc_time = log2.fold.change(
  diabetes_GeneExprs_sub_combined[,filter(diabetes_microarrays, time_h <= 36)[,"number"]], 
  diabetes_GeneExprs_sub_combined[,filter(diabetes_microarrays, time_h > 36)[,"number"]])
diabetes_fc_time
ggplot(data = diabetes_fc_time, aes(x = log2_foldchange, y = ttest_log10_pvalue, label = GeneSymbol)) + 
  geom_point() + 
  geom_text(size = 2, col  = "red")



```



