---
title: "vsnrma normalization"
author: "Bianca Greul, Carolyn Bl√ºmcke, Anna Boot, Selina Ernst"
output: html_document
---

```{r}
# Bioconductor
library(affy)
library(vsn)#
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(tidyverse)
```

```{r}
projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)
projectPath

```
PDAC dataset, GSE59761, data is downloaded from the GEO database.
Diabetes type 1 dataset, GSE53454, data is downloaded from the GEO database.

```{r}
set.seed(132) # with this vsnrma gives out the same values each time
# get CEL files
setwd(paste(projectPath, "rawdata", "GSE59761 PDAC Capan-1 Zellen", sep = "/"))
data_pancreas <- ReadAffy()
setwd(paste(projectPath, "rawdata", "GSE53454 Diabetes type 1", sep = "/"))
data_diabetes <- ReadAffy()
# change cdf
data_pancreas@cdfName <- "HGU133Plus2_Hs_ENST"
data_diabetes@cdfName <- "HGU133Plus2_Hs_ENST"
# create expression matrix
pancreas_matrix <- exprs(data_pancreas)
diabetes_matrix <- exprs(data_diabetes)
# store colnames (microarray)
pancreas_microarray_information <- colnames(pancreas_matrix)
diabetes_microarray_information <- colnames(diabetes_matrix)
# vsnrm normalization
pancreas_vsnrma <- vsnrma(data_pancreas)
diabetes_vsnrma <- vsnrma(data_diabetes)
# expression matrix of vsnrm normalized data set
pancreas_vsnrma_matrix <- exprs(pancreas_vsnrma)
diabetes_vsnrma_matrix <- exprs(diabetes_vsnrma)
# cut rownames at "." 
rownames(pancreas_vsnrma_matrix) <- str_replace(rownames(pancreas_vsnrma_matrix),"\\..*","") 
rownames(diabetes_vsnrma_matrix) <- str_replace(rownames(diabetes_vsnrma_matrix),"\\..*","")
# remove ending .CEL from colnames 
colnames(pancreas_vsnrma_matrix) <- str_remove(colnames(pancreas_vsnrma_matrix),"_.CEL")
colnames(diabetes_vsnrma_matrix) <- str_replace(colnames(diabetes_vsnrma_matrix),"\\..*","")
# transcript IDs
pancreas_transcript_names = rownames(pancreas_vsnrma_matrix)
diabetes_transcript_names = rownames(diabetes_vsnrma_matrix)
# remove control probes (AFFX) 
pancreas_vsnrma_matrix = pancreas_vsnrma_matrix[which(!startsWith(pancreas_transcript_names, "AFFX")),]
diabetes_vsnrma_matrix = diabetes_vsnrma_matrix[which(!startsWith(diabetes_transcript_names, "AFFX")),]
# check if values are equal each time code is run 
min(pancreas_vsnrma_matrix) 
# 5.223044
min(diabetes_vsnrma_matrix)
# 5.678696
```

```{r}
# create data frame with information about microarray samples
##
diabetes_microarrays = t(as.data.frame(
  sapply(diabetes_microarray_information, function(x){
    s = str_replace(x,"\\..*","")
    str_split(s, "_")
    })
  ))[,-c(2,3)]
colnames(diabetes_microarrays) <- c("number", "treatment", "time_h")
diabetes_microarrays <- transform(diabetes_microarrays, 
                                  time_h = as.integer(str_remove(diabetes_microarrays[,"time_h"],"h")))
rownames(diabetes_microarrays) <- c(1:24)
diabetes_microarrays <- as.data.frame(diabetes_microarrays)
##
pancreas_microarrays <- data.frame(number = substr(pancreas_microarray_information, 1,10), 
                                   version = substr(pancreas_microarray_information, 24,27),
                                   siRNA = c("TBL1", "TBL1", "TBL1", "NC", "NC", "NC"), 
                                   time = rep("24h",6),
                                   Affymetrix = rep("G-U133_Plus_2",6),
                                   row.names = c(1:6))


```


```{r}
# get table for converting transcript IDs to gene symbols
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
annotation = read.csv("ensemble.103.txt") 
ensemble_transcripts  = annotation[,"Transcript.stable.ID"]
ensemble_symbol = annotation[,"HGNC.symbol"]
# name each gene symbol by their respective transcripts
names(ensemble_symbol) <- ensemble_transcripts
# select only those transcripts that are also present in the annotation
diabetes_vsnrma_matrix <- diabetes_vsnrma_matrix[rownames(diabetes_vsnrma_matrix) %in% ensemble_transcripts,]
pancreas_vsnrma_matrix <- pancreas_vsnrma_matrix[rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts,]

# create vector with the symbols of the transcript from expression matrix
diabetes_symbol <- ensemble_symbol[rownames(diabetes_vsnrma_matrix)]
pancreas_symbol <- ensemble_symbol[rownames(pancreas_vsnrma_matrix)]
# use gene symbols instead of transcripts as rownames
diabetes_exprs <- cbind(GeneSymbol = as.character(diabetes_symbol),data.frame(diabetes_vsnrma_matrix))
pancreas_exprs <- cbind(GeneSymbol = as.character(pancreas_symbol),data.frame(pancreas_vsnrma_matrix))
rownames(diabetes_vsnrma_matrix) <- as.character(diabetes_symbol)
rownames(pancreas_vsnrma_matrix) <- as.character(pancreas_symbol)
# order genes alphabetically & remove all those rows where "" is the rowname 
diabetes_GeneExprs <- diabetes_vsnrma_matrix[order(rownames(diabetes_vsnrma_matrix)),][1097:nrow(diabetes_vsnrma_matrix),] 
pancreas_GeneExprs <-pancreas_vsnrma_matrix[order(rownames(pancreas_vsnrma_matrix)),][1097:nrow(pancreas_vsnrma_matrix),] 
# replace colnames with the GSM number of the microarrays
colnames(diabetes_GeneExprs) <- diabetes_microarrays[,"number"]
colnames(pancreas_GeneExprs) <- pancreas_microarrays[,"number"]
colnames(diabetes_exprs) <- c("GeneSymbol", diabetes_microarrays[,"number"])
colnames(pancreas_exprs) <- c("GeneSymbol", pancreas_microarrays[,"number"])
# as data frame (transcript IDs were kept as rownames)
diabetes_exprs <- arrange(filter(diabetes_exprs, GeneSymbol != ""), GeneSymbol)
pancreas_exprs <- arrange(filter(pancreas_exprs, GeneSymbol != ""), GeneSymbol)

```

```{r}
# get the tissue specific genes from "TRA Daten"
setwd(paste(projectPath, "sessions", "RData" ,sep = "/"))
load("TRA_pancreas_genes_human.RData") # vector with pancreas specifc TRAs 
# create a expression matrix with only those genes
diabetes_GeneExprs_sub = diabetes_GeneExprs[which(rownames(diabetes_GeneExprs) %in% TRA_pancreas_genes_human),]
pancreas_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% TRA_pancreas_genes_human),]
# same for data frame (transcript IDs were kept as rownames)
diabetes_TRAexprs <- filter(diabetes_exprs, GeneSymbol %in% TRA_pancreas_genes_human)
pancreas_TRAexprs <- filter(pancreas_exprs, GeneSymbol %in% TRA_pancreas_genes_human)
```

```{r}
combineGeneExprs_median2 <- function(exprsmatrix){
  # create data frame with a column of all the gene symbols
  data <- data.frame(geneSymbols = rownames(exprsmatrix),
                     exprsmatrix, 
                     row.names = NULL)
  # group data frame by gene symbols and combine values of one gene by median
  combined <- aggregate(data[-1], by = list(data$geneSymbols) , FUN = median)
  # convert data frame to matrix (only expression values, no row names)
  result <- as.matrix(combined[,-1])
  # use gene symbols as row names
  rownames(result) <- combined[,1]
  # output
  result
}

```


```{r}
# combine gene expression (one gene sometimes has many transcripts)
pancreas_GeneExprs_combined <- combineGeneExprs_median2(pancreas_GeneExprs)
diabetes_GeneExprs_combined <- combineGeneExprs_median2(diabetes_GeneExprs)
pancreas_GeneExprs_sub_combined <- combineGeneExprs_median2(pancreas_GeneExprs_sub)
diabetes_GeneExprs_sub_combined <- combineGeneExprs_median2(diabetes_GeneExprs_sub)
# combine all rows where values are exactly the same (leave isoforms)
pancreas_TRAexprs <- distinct(pancreas_TRAexprs)
diabetes_TRAexprs <- distinct(diabetes_TRAexprs)

```

```{r}
setwd(paste(projectPath, "sessions", "RData" ,sep = "/"))

# information about microarrays

save(pancreas_microarrays, file = "pancreas_microarrays.RData")
save(diabetes_microarrays, file = "diabetes_microarrays.RData")

# after vsnrma normalization + editing transcript IDs + removing controls (AFFX) + removing .CEL ending

save(pancreas_vsnrma_matrix, file = "pancreas_vsnrma_matrix.RData")
save(diabetes_vsnrma_matrix, file = "diabetes_vsnrma_matrix.RData")

# after replacing affyIDs (Transcripts) with gene symbols + removing "" rows

save(pancreas_GeneExprs, file = "pancreas_GeneExprs.RData")
save(diabetes_GeneExprs, file = "diabetes_GeneExprs.RData")

# after selecting only pancreas specific TRAs 

save(pancreas_GeneExprs_sub, file = "pancreas_GeneExprs_sub.RData")
save(diabetes_GeneExprs_sub, file = "diabetes_GeneExprs_sub.RData")

# after combining

save(pancreas_GeneExprs_sub_combined, file = "pancreas_GeneExprs_sub_combined.RData") 
save(diabetes_GeneExprs_sub_combined, file = "diabetes_GeneExprs_sub_combined.RData") 

save(pancreas_GeneExprs_combined, file = "pancreas_GeneExprs_combined.RData") 
save(diabetes_GeneExprs_combined, file = "diabetes_GeneExprs_combined.RData") 

# data frame with all processing steps except for combining by median 
## instead: distinct (removes duplicated rows) 

save(pancreas_TRAexprs, file = "pancreas_TRAexprs.RData")
save(diabetes_TRAexprs, file = "diabetes_TRAexprs.RData")
```



```{r}
# How to save as .csv file
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
write.csv2(diabetes_vsnrma_matrix, file = "diabetes_vsnrma_matrix.csv", sep = ";", dec = ",")

setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
write.csv2(pancreas_vsnrma_matrix, file = "pancreas_vsnrma_matrix.csv", sep = ";", dec = ",")

# reading (problem: values can differ a little from original)
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
test = read.csv2("diabetes_vsnrma_matrix.csv")
IDs = test[,"X"]
diabetes_vsnrma_matrix = as.matrix(test[,2:ncol(test)])
rownames(diabetes_vsnrma_matrix) <- IDs
remove(IDs); remove(test)

setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
test = read.csv2("pancreas_vsnrma_matrix.csv")
IDs = test[,"X"]
pancreas_vsnrma_matrix = as.matrix(test[,2:ncol(test)])
rownames(pancreas_vsnrma_matrix) <- IDs
remove(IDs); remove(test)

```
