---
title: "Analysis of PDAC data set"
date: "2 5 2021"
output: html_document
---

# libraries

```{r libraries, include=FALSE}
# Bioconductor
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)

library(rstudioapi)
library(tidyverse)
library(ggplot2)
library(pheatmap)
library(stats)
library(dbplyr)
```

# functions
## combineGeneExprs
```{r}
# ------------------------------------------------
# function: genes in expression matrix that are mentioned more than once are combined (is mean or median better?)
# ------------------------------------------------

combineGeneExprs <- function(exprsmatrix){
  
  # no genes mentioned multiple times
  a = unique(rownames(exprsmatrix)) 
  # list of indices where genes from a are mentioned
  id = sapply(a, function(x){
    which(rownames(exprsmatrix) == x)
  })  
  
  # create empty matrix 
  h = matrix(ncol = ncol(exprsmatrix), nrow = 0)
  # fill matrix with combined expression for genes from a
  for (i in c(1:length(id))){
    m = exprsmatrix[id[[i]],]
    if (length(id[[i]]) > 1){
      # combine values (mean)
      v = apply(m,2,function(x){mean(x)})
      h = rbind(h,v)
    }else if (length(id[[i]]) == 1){
      h = rbind(h,m)
    }
  }
  # change rownames to gene symbols
  rownames(h) = a
  # output
  h
}

```

# expression data sets
## create vsn normalized expression matrix
PDAC dataset, GSE59761, data is downloaded from the GEO database
```{r}
# ------------------------------------------------
# steps already done in Quality Control
# ------------------------------------------------

projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)
# get CEL files
setwd(paste(projectPath, "rawdata", "GSE59761 PDAC Capan-1 Zellen", sep = "/"))
data_pancreas <- ReadAffy()
# change cdf
data_pancreas@cdfName <- "HGU133Plus2_Hs_ENST"
# create expression matrix
pancreas_matrix <- as.data.frame(exprs(data_pancreas))
# store colnames (microarray)
pancreas_microarray_information <- colnames(pancreas_matrix)
# vsnrm normalization
pancreas_vsnrma <- vsnrma(data_pancreas)
# expression matrix of vsnrm normalized data set
pancreas_vsnrma_matrix <- exprs(pancreas_vsnrma)
# cut rownames at "." 
rownames(pancreas_vsnrma_matrix) <- str_replace(rownames(pancreas_vsnrma_matrix),"\\..*","") 

```


## information about microarray samples

```{r}
# ------------------------------------------------
# create data frame with information about microarray samples
# ------------------------------------------------

pancreas_microarrays <- data.frame(number = substr(pancreas_microarray_information, 1,10), 
                                   version = substr(pancreas_microarray_information, 24,27),
                                   siRNA = c("TBL1", "TBL1", "TBL1", "NC", "NC", "NC"), 
                                   time = rep("24h",6),
                                   Affymetrix = rep("G-U133_Plus_2",6),
                                   row.names = c(1:6))

# replace old IDs sample names
colnames(pancreas_matrix) <- pancreas_microarrays[,"number"]
colnames(pancreas_vsnrma_matrix) <- pancreas_microarrays[,"number"]
rownames(data_pancreas@phenoData) <- pancreas_microarrays[,"number"]
rownames(data_pancreas@protocolData) <- pancreas_microarrays[,"number"]

# how does the data frame look?
pancreas_microarrays
```


# Processing of data
## remove control probes

```{r}
# ------------------------------------------------
# remove control probes from normalized data set
# ------------------------------------------------
dim(pancreas_vsnrma_matrix) # before
# 95721     6

pancreas_transcript_names = rownames(pancreas_vsnrma_matrix)
pancreas_vsnrma_matrix = pancreas_vsnrma_matrix[which(!startsWith(pancreas_transcript_names, "AFFX")),] # remove control probes (AFFX) 

dim(pancreas_vsnrma_matrix) # after
# 95659     6
```
## convert transcript IDs to gene symbols
```{r}
# ------------------------------------------------
# convert Affymetrix IDs to gene symbols
# ------------------------------------------------

setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
annotation = read.csv("ensemble.103.txt") 
# file comes from www.ensembl.org/biomart

```


```{r}
ensemble_genes = annotation[,"Gene.stable.ID"]
ensemble_transcripts  = annotation[,"Transcript.stable.ID"]
ensemble_affyID = annotation[,"AFFY.HG.U133.Plus.2.probe"]
ensemble_symbol = annotation[,"HGNC.symbol"]

# which transcripts from the microarray are not listed in the ensemble.103.txt file?
pancreas_nomatchesIDs <- rownames(pancreas_vsnrma_matrix)[!(rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts)]
length(pancreas_nomatchesIDs)
# 112

# remove these 112 transcripts from expression data
pancreas_vsnrma_matrix <- pancreas_vsnrma_matrix[rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts,]
dim(pancreas_vsnrma_matrix)
# 95547     6

# ------------------------------------------------
# select only those symbols that are in the pancreatic cancer data set
# ------------------------------------------------

names(ensemble_symbol) <- ensemble_transcripts 
pancreas_symbol <- ensemble_symbol[rownames(pancreas_vsnrma_matrix)]
# symbols already in right order for pancreas_vsnrma_matrix

# ------------------------------------------------
# name the probes according to gene symbol
# ------------------------------------------------

rownames(pancreas_vsnrma_matrix) <- as.character(pancreas_symbol)

# ------------------------------------------------
# check for NAs or "" 
# ------------------------------------------------

#apply(pancreas_vsnrma_matrix, 2, function(x){sum(x == 0)})

sum(sapply(rownames(pancreas_vsnrma_matrix), function(x){sum(is.na(x))})) # 0 are NA

sum(sapply(rownames(pancreas_vsnrma_matrix), function(x){sum(x == "", na.rm = TRUE)})) # 1096 are ""

```

## sort gene symbols alphabetically 
```{r}
# ------------------------------------------------
# alphabetical order + deletion of all rows with "" as gene symbol (1096)
# ------------------------------------------------

pancreas_GeneExprs <-pancreas_vsnrma_matrix[order(rownames(pancreas_vsnrma_matrix)),][1097:nrow(pancreas_vsnrma_matrix),] 

dim(pancreas_GeneExprs)
# 94451     6

head(pancreas_GeneExprs)
```


```{r}
# ------------------------------------------------
# save as .csv file 
# ------------------------------------------------

# create excel table with columns: genenames, gene expression values for each mircoarray
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
write.csv2(pancreas_GeneExprs, file = "pancreas_GeneExprs_allgenes.csv", sep = ";", dec = ",", col.names = NA, row.names = TRUE)
# can be read by using read.csv2
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
test = read.csv2("pancreas_GeneExprs_allgenes.csv", col.names = c("GeneSymbols", pancreas_microarrays[,"number"]))
sym = test[,"GeneSymbols"]
test = as.matrix(test[,pancreas_microarrays[,"number"]])
rownames(test) <- sym 
# some smaller changes made to the actual values, but still nearly equal
all.equal(test,pancreas_GeneExprs)
# variable test is not really needed
remove(sym)
remove(test)

```

## select only pancreas specific TRAs
```{r}
# ------------------------------------------------
# minimize expression matrix: only pancreas TRAs 
# ------------------------------------------------

# get the tissue specifc genes from TRA Daten
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
pancreas_specific_genes = read.csv("pancreas_specific_genes_human.csv")
pancreas_specific_genes = sort(pancreas_specific_genes) 

# create a expression matrix with only those genes
pancreas_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% pancreas_specific_genes),]


length(unique(rownames(pancreas_GeneExprs_sub)))
# 250
dim(pancreas_GeneExprs_sub)
# 1569    6
head(pancreas_GeneExprs_sub)

```

## combine genes
```{r}
# ------------------------------------------------
# use of own function to combine values of genes that are mentioned more than once
# ------------------------------------------------

dim(combineGeneExprs(pancreas_GeneExprs_sub))
# 250  6

dim(combineGeneExprs(pancreas_GeneExprs))
# 18062    6
```
## select only all TRAs
```{r}
# ------------------------------------------------
# minimize expression matrix: only all TRAs 
# ------------------------------------------------

# get TRA gene Daten
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
all_TRA_genes = read.csv("TRA_genes_human.csv")
all_TRA_genes = sort(all_TRA_genes) 

# create a expression matrix with only those genes
all_TRA_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% all_TRA_genes),]


length(unique(rownames(all_TRA_GeneExprs_sub)))
dim(all_TRA_GeneExprs_sub)
head(all_TRA_GeneExprs_sub)

```

```{r}
## combine
#funktioniert bei mir hier nicht
# ------------------------------------------------
# use of own function to combine values of genes that are mentioned more than once
# ------------------------------------------------

dim(combineGeneExprs(all_TRA_GeneExprs_sub))
# 14388     6 

# dim(combineGeneExprs(all_TRA_GeneExprs)) # WHAT is all_TRA_GeneExprs?

```


# plots
## distributions

```{r fig.height=12}
par(mfrow = c(2,3), mai = c(0.6,0.6,0.5,0.1))

for(x in 1:6){
  hist(pancreas_GeneExprs[,x], 
       xlab = "gene expression values", cex.main = 2, cex.lab = 1.5,
       main = paste("Histogram of", pancreas_microarrays[x,"number"],
                    "\n", "data set: GSE59761"))
  abline(v = c(mean(pancreas_GeneExprs[,x]), median(pancreas_GeneExprs[,x])), 
         col = c("red", "blue"), lwd = 2)
  legend("topright", legend = c("mean", "median"), fill = c("red", "blue"), cex = 1.5)
}

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_histogram_microarrays.pdf")

```

## heatmap of TBL1X gene
```{r}
# check if the knockout of TBL1X can be seen
par(mai = c(1,0.1,2,1))
pheatmap(pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) == "TBL1X"),], 
         labels_col = paste(pancreas_microarrays[,"number"], 
                            "\n siRNA:", pancreas_microarrays[,"siRNA"]),
         main = "TBL1X - Gene expression in pancreatic cancer cells\ndata set: GSE59761 (Stoy et al., 2015)",
         cluster_rows = TRUE, treeheight_row = 0, 
         fontsize_row = 8, fontsize_col = 7, fontsize = 9)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_heatmap_TBL1X.pdf")


```


## heatmap of TRAs

```{r}

pheatmap(pancreas_GeneExprs_sub,
         labels_col = paste(pancreas_microarrays[,"number"],"\n", "siRNA:", pancreas_microarrays[,"siRNA"]),
         labels_row = rep("",nrow(pancreas_GeneExprs_sub)),
         cluster_rows = TRUE, treeheight_row = 0,
         fontsize_row = 8, fontsize_col = 7, fontsize = 9,
         main = "Heatmap of TRA gene expression in pancreatic cancer cells \n data set: GSE59761")

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_heatmap_allTRAs.pdf")
```


## heatmap of TRAs
```{r fig.height=12}
# ------------------------------------------------
# boxplot (very unfinished, just to get an overview)
# ------------------------------------------------

par(las = 2)

boxplot(t(combineGeneExprs(pancreas_GeneExprs_sub)),
         col = rainbow(250),
         main="",
         horizontal = TRUE)
abline(v =  median(apply(pancreas_vsnrma_matrix, 1,median)))


```


