---
title: "Analysis of PDAC data set"
date: "2 5 2021"
output: html_document
---

# libraries

```{r libraries, include=FALSE}
# Bioconductor
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)

library(rstudioapi)
library(tidyverse)
library(ggplot2)
library(pheatmap)
library(stats)
library(dbplyr)
library(ggfortify) #new
library(reshape2) #new
library(ggrepel) #new
```

# functions
## combineGeneExprs
```{r}
# ------------------------------------------------
# function: genes in expression matrix that are mentioned more than once are combined (is sum or median better?)
# ------------------------------------------------

combineGeneExprs_median <- function(exprsmatrix){
  # all unique genes
  genes <- unique(rownames(exprsmatrix))
  # empty matrix
  result <- matrix(ncol = ncol(exprsmatrix), nrow = 0)
  # for each gene: find indices, combine rows by using median
  for (i in genes){
    indices <- which(rownames(exprsmatrix) == i)
    gene_matrix <- exprsmatrix[indices,]
    # combine values only if gene_matrix has 2 or more rows
    if (length(indices) > 1){
      # for each microarray: compute median of gene expression for certain gene 
      combined <-  apply(gene_matrix,2,function(x){median(x)})
      result <- rbind(result, combined, deparse.level = 1)}
    # if there is only one row: no need to combine values
    else if (length(indices) == 1){result = rbind(result, gene_matrix)}
  }
  rownames(result) = genes
  # output
  result
}

combineGeneExprs_sum <- function(exprsmatrix){
  # all unique genes
  genes <- unique(rownames(exprsmatrix))
  # empty matrix
  result <- matrix(ncol = ncol(exprsmatrix), nrow = 0)
  # for each gene: find indices, combine rows by using sum
  for (i in genes){
    indices <- which(rownames(exprsmatrix) == i)
    gene_matrix <- exprsmatrix[indices,]
    # combine values only if gene_matrix has 2 or more rows
    if (length(indices) > 1){
      # for each microarray: compute sum of gene expression for certain gene 
      combined <-  apply(gene_matrix,2,function(x){sum(x)})
      result <- rbind(result, combined, deparse.level = 1)}
    # if there is only one row: no need to combine values
    else if (length(indices) == 1){result = rbind(result, gene_matrix)}
  }
  rownames(result) = genes
  # output
  result
}

combineGeneExprs_median2 <- function(exprsmatrix){
  # create data frame with a column of all the gene symbols
  data <- data.frame(geneSymbols = rownames(exprsmatrix),
                     exprsmatrix, 
                     row.names = NULL)
  # group data frame by gene symbols and combine values of one gene by median
  combined <- aggregate(data[-1], by = list(data$geneSymbols) , FUN = median)
  # convert data frame to matrix (only expression values, no row names)
  result <- as.matrix(combined[,-1])
  # use gene symbols as row names
  rownames(result) <- combined[,1]
  # output
  result
}


```

# expression data sets
## create vsn normalized expression matrix
PDAC dataset, GSE59761, data is downloaded from the GEO database
```{r}

projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)
# get CEL files
setwd(paste(projectPath, "rawdata", "GSE59761 PDAC Capan-1 Zellen", sep = "/"))
data_pancreas <- ReadAffy()
# change cdf
data_pancreas@cdfName <- "HGU133Plus2_Hs_ENST"
# create expression matrix
pancreas_matrix <- exprs(data_pancreas)
# store colnames (microarray)
pancreas_microarray_information <- colnames(pancreas_matrix)
# vsnrm normalization
pancreas_vsnrma <- vsnrma(data_pancreas)
# expression matrix of vsnrm normalized data set
pancreas_vsnrma_matrix <- exprs(pancreas_vsnrma)
# cut rownames at "."
rownames(pancreas_vsnrma_matrix) <- str_replace(rownames(pancreas_vsnrma_matrix),"\\..*","")


# # get expression matrix 
# # steps already done: vsnrma normalization + editing transcript IDs + removing controls (AFFX) + removing .CEL ending
# setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
# test = read.csv2("pancreas_vsnrma_matrix.csv")
# IDs = test[,"X"]
# pancreas_vsnrma_matrix = as.matrix(test[,2:ncol(test)])
# rownames(pancreas_vsnrma_matrix) <- IDs
# remove(IDs); remove(test)

```


## information about microarray samples

```{r}
# ------------------------------------------------
# create data frame with information about microarray samples
# ------------------------------------------------

pancreas_microarrays <- data.frame(number = substr(pancreas_microarray_information, 1,10), 
                                   version = substr(pancreas_microarray_information, 24,27),
                                   siRNA = c("TBL1", "TBL1", "TBL1", "NC", "NC", "NC"), 
                                   time = rep("24h",6),
                                   Affymetrix = rep("G-U133_Plus_2",6),
                                   row.names = c(1:6))

# replace old IDs sample names
colnames(pancreas_matrix) <- pancreas_microarrays[,"number"]
colnames(pancreas_vsnrma_matrix) <- pancreas_microarrays[,"number"]
rownames(data_pancreas@phenoData) <- pancreas_microarrays[,"number"]
rownames(data_pancreas@protocolData) <- pancreas_microarrays[,"number"]

# how does the data frame look?
pancreas_microarrays
```


# Processing of data
## remove control probes

```{r}
# ------------------------------------------------
# remove control probes from normalized data set
# ------------------------------------------------
dim(pancreas_vsnrma_matrix) # before
# 95721     6

pancreas_transcript_names = rownames(pancreas_vsnrma_matrix)
pancreas_vsnrma_matrix = pancreas_vsnrma_matrix[which(!startsWith(pancreas_transcript_names, "AFFX")),] # remove control probes (AFFX) 

dim(pancreas_vsnrma_matrix) # after
# 95659     6

```

## convert transcript IDs to gene symbols

```{r}
# ------------------------------------------------
# convert Affymetrix IDs to gene symbols
# ------------------------------------------------

setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
annotation = read.csv("ensemble.103.txt") 
# file comes from www.ensembl.org/biomart

```


```{r}
ensemble_genes = annotation[,"Gene.stable.ID"]
ensemble_transcripts  = annotation[,"Transcript.stable.ID"]
ensemble_affyID = annotation[,"AFFY.HG.U133.Plus.2.probe"]
ensemble_symbol = annotation[,"HGNC.symbol"]

# which transcripts from the microarray are not listed in the ensemble.103.txt file?
pancreas_nomatchesIDs <- rownames(pancreas_vsnrma_matrix)[!(rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts)]
length(pancreas_nomatchesIDs)
# 112

# remove these 112 transcripts from expression data
pancreas_vsnrma_matrix <- pancreas_vsnrma_matrix[rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts,]
dim(pancreas_vsnrma_matrix)
# 95547     6


# select only those symbols that are in the pancreatic cancer data set
names(ensemble_symbol) <- ensemble_transcripts 
pancreas_symbol <- ensemble_symbol[rownames(pancreas_vsnrma_matrix)]
# symbols already in right order for pancreas_vsnrma_matrix


# name the probes according to gene symbol
rownames(pancreas_vsnrma_matrix) <- as.character(pancreas_symbol)

# ------------------------------------------------
# check for NAs or "" 
# ------------------------------------------------

sum(sapply(rownames(pancreas_vsnrma_matrix), function(x){sum(is.na(x))})) 
# 0 are NA

sum(sapply(rownames(pancreas_vsnrma_matrix), function(x){sum(x == "", na.rm = TRUE)})) 
# 1096 are "" 

```

## sort gene symbols alphabetically 
```{r}
# ------------------------------------------------
# alphabetical order + deletion of all rows with "" as gene symbol (1096)
# ------------------------------------------------

pancreas_GeneExprs <-pancreas_vsnrma_matrix[order(rownames(pancreas_vsnrma_matrix)),][1097:nrow(pancreas_vsnrma_matrix),] 

dim(pancreas_GeneExprs)
# 94451     6

head(pancreas_GeneExprs)
```


```{r}
# ------------------------------------------------
# save as .csv file 
# ------------------------------------------------

# create excel table with columns: genenames, gene expression values for each mircoarray
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
write.csv2(pancreas_GeneExprs, file = "pancreas_GeneExprs.csv", sep = ";", dec = ",")

# # can be read by using read.csv2
# setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
# test = read.csv2("pancreas_GeneExprs.csv")
# sym = test[,"X"]
# pancreas_GeneExprs = as.matrix(test[,2:ncol(test)])
# rownames(pancreas_GeneExprs) <- sym 
# pancreas_GeneExprs
# remove(sym); remove(test)
```

## select only pancreas specific TRAs
```{r}
# ------------------------------------------------
# minimize expression matrix: only pancreas TRAs 
# ------------------------------------------------

# get the tissue specifc genes from TRA Daten
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
pancreas_specific_genes = read.csv("pancreas_specific_genes_human.csv")
pancreas_specific_genes = sort(pancreas_specific_genes) 

# create a expression matrix with only those genes
pancreas_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% pancreas_specific_genes),]


length(unique(rownames(pancreas_GeneExprs_sub)))
# 250
dim(pancreas_GeneExprs_sub)
# 1569    6
head(pancreas_GeneExprs_sub)

```

## combine genes
```{r}
# ------------------------------------------------
# use of own function to combine values of genes that are mentioned more than once
# ------------------------------------------------

pancreas_GeneExprs_sub_combined <- combineGeneExprs_median2(pancreas_GeneExprs_sub)
dim(combineGeneExprs_median2(pancreas_GeneExprs_sub))
# 250  6

dim(combineGeneExprs_median2(pancreas_GeneExprs))
# 18062    6
```
## select only all TRAs
```{r}
# ------------------------------------------------
# minimize expression matrix: only all TRAs 
# ------------------------------------------------

# get TRA gene Daten
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
all_TRA_genes = read.csv("TRA_genes_human.csv")
all_TRA_genes = sort(all_TRA_genes) 

# create a expression matrix with only those genes
all_TRA_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% all_TRA_genes),]


length(unique(rownames(all_TRA_GeneExprs_sub)))
dim(all_TRA_GeneExprs_sub)
head(all_TRA_GeneExprs_sub)

```

```{r}
## combine
#funktioniert bei mir hier nicht
# ------------------------------------------------
# use of own function to combine values of genes that are mentioned more than once
# ------------------------------------------------

dim(combineGeneExprs_median2(all_TRA_GeneExprs_sub))
# 14388     6 

```


# plots
## distributions

```{r fig.height=12}
par(mfrow = c(2,3), mai = c(0.6,0.6,0.5,0.1))

for(x in 1:6){
  hist(pancreas_GeneExprs[,x], 
       xlab = "gene expression values", cex.main = 2, cex.lab = 1.5,
       main = paste("Histogram of", pancreas_microarrays[x,"number"],
                    "\n", "data set: GSE59761"))
  abline(v = c(mean(pancreas_GeneExprs[,x]), median(pancreas_GeneExprs[,x])), 
         col = c("red", "blue"), lwd = 2)
  legend("topright", legend = c("mean", "median"), fill = c("red", "blue"), cex = 1.5)
}

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_histogram_microarrays.pdf")

```

## heatmap of TBL1X gene
```{r}
# check if the knockout of TBL1X can be seen
par(mai = c(1,0.1,2,1))
pheatmap(pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) == "TBL1X"),], 
         labels_col = paste(pancreas_microarrays[,"number"], 
                            "\n siRNA:", pancreas_microarrays[,"siRNA"]),
         main = "TBL1X - Gene expression in pancreatic cancer cells\ndata set: GSE59761 (Stoy et al., 2015)",
         cluster_rows = TRUE, treeheight_row = 0, 
         fontsize_row = 8, fontsize_col = 7, fontsize = 9)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_heatmap_TBL1X.pdf")


```


## heatmap of TRAs

```{r}

pheatmap(pancreas_GeneExprs_sub,
         labels_col = paste(pancreas_microarrays[,"number"],"\n", "siRNA:",
                            pancreas_microarrays[,"siRNA"]),
         labels_row = rep("",nrow(pancreas_GeneExprs_sub)),
         cluster_rows = TRUE, treeheight_row = 0,
         fontsize_row = 8, fontsize_col = 7, fontsize = 9,
         main = "Heatmap of TRA gene expression in pancreatic cancer cells \n data set: GSE59761")

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_heatmap_allTRAs.pdf")
```


## heatmap of TRAs
```{r fig.height=12}
# ------------------------------------------------
# boxplot (very unfinished, just to get an overview)
# ------------------------------------------------

par(las = 2)

boxplot(t(pancreas_GeneExprs_sub_combined),
         col = rainbow(250),
         main="",
         horizontal = TRUE)
abline(v =  median(apply(pancreas_vsnrma_matrix, 1,median)))


```


# control vs. treatment

## PCA & kmeans

```{r}
# ------------------------------------------------
# PCA & kmeans
# ------------------------------------------------

# create a data frame (columns = treatment, median combined pancreas specifc TRAs)
pancreas_df <-cbind(treatment = pancreas_microarrays[,"siRNA"],
            data.frame(t(pancreas_GeneExprs_sub_combined)))
head(pancreas_df) 

# PCA on the TRA expressions
pancreas_pca <- prcomp(pancreas_df[,-1], scale. = TRUE)

# elbow plot 
plot(x = 1:5, y = sapply(1:5, function(i){kmeans(pancreas_pca$x,i)$tot.withinss}), 
     type = "l", xlab = "number of clusters", ylab = "total within-cluster sum of squares")
# result: k = 2 

# plot PC1 against PC2 and show the eigenvectors (loadings)
autoplot(pancreas_pca, data = pancreas_df, colour = "treatment", main = "Principal Component Analysis on PDAC data set (GSE59761)", xlim = c(-0.7,0.7), ylim = c(-0.7,0.7), asp = 1)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_pca_microarrays.pdf")

autoplot(pancreas_pca, data = pancreas_df, colour = "treatment", loadings = TRUE, loadings.label = TRUE, loadings.label.size = 3, main = "Principal Component Analysis on PDAC data set (GSE59761)", xlim = c(-0.7,0.7), ylim = c(-0.7,0.7), asp = 1)


# plot kmeans for k = 2
autoplot(kmeans(pancreas_pca$x,2), data = pancreas_pca$x, frame = TRUE,  xlim = c(-0.7,0.7), ylim = c(-0.7,0.7), asp = 1, main = "kmeans clustering (k=2)\nafter PCA on PDAC data set (GSE59761)") + 
  scale_color_hue(direction = -1, h.start = 90) +
  scale_fill_hue(direction = -1, h.start = 90)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_pca_kmeans_microarrays.pdf")

# add cluster numbers to already existing data frame (rows = microarrays)
pancreas_df <- cbind(kmeans2 = kmeans(pancreas_pca$x,2)$cluster, pancreas_df)
head(pancreas_df)

```

```{r}
# plot eigenvector values (PC1 vs PC2)
ggplot(pancreas_pca$rotation, aes(PC1, PC2, label = rownames(pancreas_pca$rotation))) + 
  geom_point() + 
  geom_label_repel(max.overlaps = 50, size = 2) + 
  coord_equal(xlim = c(-0.7,0.7), ylim = c(-0.7,0.7))

# select the genes that contribute the most to PC1 & PC2
treshhold1 = 0.1
treshhold2 = 0.15
pancreas_pca_importantGenes = rownames(filter(data.frame(pancreas_pca$rotation), PC1 > treshhold1 | PC1 < -treshhold1 | PC2 > treshhold2 | PC2 < -treshhold2 ))

# plot only the genes that strongly influence PC1 & PC2
ggplot(pancreas_pca$rotation[pancreas_pca_importantGenes,], aes(PC1, PC2, label = pancreas_pca_importantGenes)) + 
  geom_point() + 
  geom_label_repel(max.overlaps = 50, size = 2) + 
  coord_equal(xlim = c(-0.7,0.7), ylim = c(-0.7,0.7))
```

## high variance
```{r}
# ------------------------------------------------
# find genes with high variance between the microarrays 
# (can be caused by different treatment or different time point)
# ------------------------------------------------

# compute variance of each gene (each gene: 6 values)
pancreas_var = sort(apply(pancreas_GeneExprs_sub_combined,1,var), decreasing = TRUE)

# are these high variance genes the same as the mediangenes that strongly contribute to PCA?
pancreas_pca_importantGenes %in% names(pancreas_var[pancreas_var > 2*median(pancreas_var)])
# many PCA contributing genes have a 2x variance higher than median variance
names(pancreas_var)[1:5] %in% pancreas_pca_importantGenes
# all top 5 high variance genes contribute strongly in PCA
# (there are genes with high variance that don't really contribute to PC1/PC2)
```

```{r}
# look at gene with highest variance (CHST4)
pancreas_CHST4 <- data.frame(pancreas_microarrays[,c("number","siRNA")], CHST4 = pancreas_GeneExprs_sub_combined["CHST4",])
pancreas_CHST4
ggplot(pancreas_CHST4, aes(x = siRNA, y = CHST4)) + 
  geom_boxplot()

```

## genes that differ with treatment / time
```{r}
# ------------------------------------------------
# boxplot of high variance genes (grouped by treatment / time)
# ------------------------------------------------

# melt expression values of top10 high variance genes
mel = melt(t(pancreas_GeneExprs_sub_combined[names(pancreas_var)[1:10],]))
# vector to annotate the values from mel with treatment 
tr = pancreas_microarrays[,"siRNA"]
names(tr) = pancreas_microarrays[,"number"]
tr = unname(tr[as.character(mel$Var1)])

# create data frame (used for ggplot)
pancreas_highvariance_GeneExprs <- cbind(treatment = tr, mel)
remove(tr,mel)
# plot boxplots of top10 high variance genes (grouped by treatment)
ggplot(pancreas_highvariance_GeneExprs, aes(x = Var2, y = value, colour = treatment)) + 
  geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90, size = 5))+
  xlab("Genes")

```

## statistical tests
```{r}
# ------------------------------------------------
# ttest (on gene expression of top 10 high variance genes) +
# two sided wilcoxon rank sum exact test 
# ------------------------------------------------
## H1 (treatment): treatment does not change gene expression
## H1 -> alpha = 5% (p < alpha)
## H1.15 -> alpha = 15% (p < alpha)

stat_tests <- function(df, ctrl, treat){
  
  w = apply(df, 1, function(x){
    wilcox.test(x[ctrl], x[treat])$p.value
  })
  
  t = apply(df, 1, function(x){
    t.test(x[ctrl], x[treat])$p.value
  })
  
  data.frame(wilcox_pvalue = w, 
             H1_wilcox = w < 0.05, 
             H1.15_wilcox = w < 0.15,
             ttest_pvalue = t, 
             H1_t = t < 0.05)
}


pancreas_highvariance_stat <- stat_tests(pancreas_GeneExprs_sub_combined[names(pancreas_var)[1:10],],
           pancreas_microarrays[,"siRNA"] == "NC", 
           pancreas_microarrays[,"siRNA"] == "TBL1")
pancreas_stat <- stat_tests(pancreas_GeneExprs_sub_combined, 
           pancreas_microarrays[,"siRNA"] == "NC", 
           pancreas_microarrays[,"siRNA"] == "TBL1")

filter(pancreas_highvariance_stat, H1.15_wilcox == TRUE)
filter(pancreas_highvariance_stat, H1_t == TRUE)
# each gene is contributing to PCA (part of pancreas_pca_importantGenes)
filter(pancreas_stat, H1.15_wilcox == TRUE)
filter(pancreas_stat, H1_t == TRUE)
# when H1_t = TRUE then H1.15_wilcox = TRUE (maybe just use ttest)

# plot genes with signigficant difference between treatment & control
autoplot(pancreas_GeneExprs_sub_combined[c("CHST4", "PABPC4", "ECI2", "TMEM97", "ANXA4", "NEURL3", "MKNK1", "MED24", "GRB10"),], xlab = "Microarrays", ylab = "Genes")
```

## log2 fold change

```{r}
log2.fold.change <- function(control, test){
  
  # only if control have the same row numbers
  if(nrow(control) == nrow(test)){
    
    # control / treatment (values combined by median)
    ratio = sapply(1:nrow(control), function(x){
      g1_m = median(as.numeric(control[x,]))
      g2_m = median(as.numeric(test[x,]))
      g2_m/g1_m # change of test to control
    })
    # log2 on ratio => log2(treatment/control)
    log2_foldchange = log2(ratio)
    # ttest control vs treatment (expression values)
    t = sapply(1:nrow(control), function(x){
      g1 = as.numeric(control[x,])
      g2 = as.numeric(test[x,])
      t.test(control[x,], test[x,])$p.value
    })
    # wilcoxon rank sum test control vs treatment (expression values)
    w = sapply(1:nrow(control), function(x){
      g1 = as.numeric(control[x,])
      g2 = as.numeric(test[x,])
      wilcox.test(control[x,], test[x,])$p.value
    })
    
  }else{print("Something has gone wrong with the dimensions of the groups (probably row numbers)!")}
   
  trend_t = 1:length(t)
  trend_t[t < 0.05 & log2_foldchange > 0] = "up"
  trend_t[t < 0.05 & log2_foldchange < 0] = "down"
  trend_t[t >= 0.05] = "not significant"
  
  trend_w = 1:length(w)
  trend_w[w < 0.05 & log2_foldchange > 0] = "up"
  trend_w[w < 0.05 & log2_foldchange < 0] = "down"
  trend_w[w >= 0.05] = "not significant"
  
  # output
  data.frame(GeneSymbol = rownames(control), # genes
             foldchange_ratio = ratio, # without log2
             log2_foldchange = log2_foldchange,  # log2 foldchange
             ttest_raw_pvalue = t,# ttest
             ttest_log10_pvalue = -1*log10(t),
             ttest_trend = trend_t, 
             wilcox_raw_pvalue = w, # wilcoxon rank sum test
             wilcox_log10_pvalue = -1*log10(w),
             wilcox_trend = trend_w)

}

# gene expression matrix for each subgroup (treatment, control)
pancreas_ctrl = pancreas_GeneExprs_sub_combined[,filter(pancreas_microarrays, siRNA == "NC")[,"number"]]
pancreas_treat = pancreas_GeneExprs_sub_combined[,filter(pancreas_microarrays, siRNA == "TBL1")[,"number"]]


pancreas_fc = log2.fold.change(pancreas_ctrl, pancreas_treat)
pancreas_fc

filter(pancreas_fc, abs(log2_foldchange) > 0.15)
max(abs(pancreas_fc$log2_foldchange))

ggplot(data = pancreas_fc, aes(x = log2_foldchange, y = wilcox_log10_pvalue, label = GeneSymbol, col = wilcox_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") + 
  labs(title = "Volcano plot (wilcoxon rank sum test) for PDAC data set (GSE59761)", 
       x = "log2 fold change", y = "-log10 (p-value)", colour = "trend",
       subtitle = "ratio = treatment/control") +
  xlim(c(-max(abs(pancreas_fc$log2_foldchange)),
         max(abs(pancreas_fc$log2_foldchange))))+
  scale_color_manual(values = c("grey"))+ 
  geom_text_repel(size = 2, max.overlaps = 20)

setwd(paste(projectPath, "plots", sep = "/"))
dev.copy2pdf(file = "pancreas_vulcano_wilcox.pdf")

ggplot(data = pancreas_fc, aes(x = log2_foldchange, y = ttest_log10_pvalue, label = GeneSymbol, col = ttest_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") + 
  labs(title = "Volcano plot (t test)", 
       x = "log2 fold change", y = "-log10 (p-value)") +
  xlim(c(-max(abs(pancreas_fc$log2_foldchange)),
         max(abs(pancreas_fc$log2_foldchange))))+ 
  scale_color_manual(values = c("navyblue", "grey", "orange"))


```


