---
title: "TRA expression in pancreas"
author: "Bianca Greul, Carolyn Blümcke, Anna Boot, Selina Ernst"
subtitle: Analysis of TRA expression in pancreatic cancer and diabetes type 1
abstract:
output:
  pdf_document:
    keep_tex: yes
    toc: no
    toc_depth: 2
    number_sections: yes
    fig_caption: yes
  html_document:
    toc: no
    toc_depth: '2'
    df_print: paged
documentclass: article
papersize: a4
fontsize: 12pt
geometry: left = 2 cm, right = 2 cm, top = 2 cm, bottom = 2 cm, headsep = 20pt
secnumdepth: 2
linestretch: 1
out.width: 60%
fig.align: center
header-includes:
  \usepackage{float} 
  \usepackage{graphicx} 
  \usepackage[T1]{fontenc}
  \usepackage[greek, ngerman, main=english]{babel} 
  \usepackage{helvet} 
  \renewcommand{\familydefault}{\sfdefault}
  \usepackage{titlesec} 
  \headsep = 5mm
  \pagenumbering{gobble}
  \usepackage[margin=10pt,font=small,labelfont=bf,textfont=it,labelsep=colon,width=.75\textwidth]{caption}
  \usepackage{booktabs}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  eval = TRUE,
	echo = FALSE, # source code
	collapse = FALSE, # ouput from one chunk are separate
	message = FALSE, # messages not shown in output
	warning = FALSE, # warnings not in output
	comment = "", # line before output (default: ##)
	include = FALSE, # output  
	out.width = "70%", # plot smaller
	fig.align = "center", # plot in center
	highlight = TRUE
)

```
```{r libraries}
# Bioconductor
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)

library(rstudioapi) 
library(tidyverse) 
library(ggplot2) 
library(pheatmap)
library(stats)
library(dplyr)
library(ggfortify)
library(reshape2)
library(ggrepel) 
library(RColorBrewer)
library(factoextra)
library(VennDiagram)
library(kableExtra)
library(knitr)
```


```{r}
# projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)

# used because knitting sometimes has a problem here:
projectPath <- "C:/Users/Selina Ernst/Documents/GitHub/2021-topic-02-team-04"
```

\newpage
\pagenumbering{arabic} 

# Introduction

In this project the expression of tissue restricted antigens (TRAs) in pancreatic cells is examined, concentrating on two cell types: $\beta$-cells and pancreatic ductal adenocarcinoma (PDAC) cells. \
TRAs are highly expressed in specific tissues and in the thymus, where they have part in establishing the central tolerance of the immune system. A malfunction in the negative selection of developing T-cells can lead to autoimmune diseases, like diabetes type 1 (Kyewski & Klein, 2006).\
Here autoreactive T-cells attack $\beta$-cells that, as part of the endocrine pancreatic islet, synthesize and secrete the hormone insulin (Jingli \textit{et al.}, 2020).\
Another cause is the release of proinflammatory cytokines which, among mediating the innate immune response, can activate apoptotic pathways (Clark \textit{et al.}, 2017). Together this leads to the elimination of $\beta$-cells and results in insulin deficiency and hyperglycemia. \
The other cell type are PDAC cells. Pancreatic adenocarcinoma is the most common pancreatic cancer and originates from epithelial cells from the exocrine component producing digestive enzymes. Mutations of the oncogenes and the tumor-suppressor genes stimulate cell growth, division, and survival pathways (Wolfgang \textit{et al.} 2013). \
With a 5-year survival rate of around 5% pancreatic cancer is the 4th leading cause of death by cancer in Western societies (Siegel \textit{et al.}, 2015). Therefore, identifying highly expressed TRAs that are, compared to healthy and stressed $\beta$-cells, unique to PDAC could be used to facilitate the diagnosis and development of new treatment options. Additionally, pathways leading to pancreatic cancer and diabetes were analyzed and differences established. 

# Obtaining and processing of data
## Pancreas-specifc TRA genes 

The base of the project are TRAs that are significantly higher expressed in the pancreas compared to other tissues. \
The first step of was therefore to find pancreas-specific genes. The tables used were provided by Dr. Dinkelacker and came from multiple sources, all of which focused on TRA expression in human and mice tissues. By selecting all genes that were associated with the pancreas, 330 pancreas-specific TRAs in the human and 59 in mice could be gathered.\
As the project was about TRA expression in human pancreatic cells, the focus of the analysis layed on the human pancreas-specific TRAs found.

```{r}
# ----------------------------------------------------------
# TRAs 
# ----------------------------------------------------------

# set working directory to folder with TRA tables (rawdata/TRA Daten)
setwd(paste(projectPath, "rawdata", "TRA Daten", sep = "/"))
# read in all 6 tables
TRAs1_human <-read_csv("Human_protein_atlas_TRA_5median_genes_annotated.tsv")
TRAs2_human <-read_tsv("tra.2014.human.5x.table.tsv")
TRAs3_human <-read_tsv("tra.2014.human.roth.5x.table.tsv")
TRAs4_mouse <-read_tsv("tra.2014.mouse.5x.table.tsv")
TRAs5_mouse <-read_tsv("tra.2014.mouse.4301.5x.table.tsv")
TRAs6_human <-read_tsv("tra.2017.human.gtex.5x.table.tsv",col_types = cols(ensembl.chrom=col_character()))
#col_types to correct the parsing failure

# 
tissuecount <- function(df, maxtissue){
  tissues <- table(df[,maxtissue])
  percentage <- tissues/sum(tissues)
  m <- matrix(c(tissues, percentage), ncol = 2)
  rownames(m) <- names(tissues)
  colnames(m) <- c("Anzahl TRAs in Gewebe", "Anteil")
  m <- m[order(m[,1], decreasing = TRUE),]
  m
}

# ----------------------------------------------------------
# distribution of TRAs betweeen tissues
# ----------------------------------------------------------

GewebeTabelleProzent1 <- tissuecount(TRAs1_human,"Max_tissue")
GewebeTabelleProzent2 <- tissuecount(TRAs2_human,"max.tissue")
GewebeTabelleProzent3 <- tissuecount(TRAs3_human,"max.tissue")
GewebeTabelleProzent4 <- tissuecount(TRAs4_mouse,"max.tissue")
GewebeTabelleProzent5 <- tissuecount(TRAs5_mouse,"max.tissue")
GewebeTabelleProzent6 <- tissuecount(TRAs6_human,"max.tissue")

# ----------------------------------------------------------
# Pancreas specific TRAs
# ----------------------------------------------------------

# create data frames where only pancreas genes are listed
PancreasTRAs1_human <- TRAs1_human[grep(x=TRAs1_human$Max_tissue,"panc"),]
PancreasTRAs2_human <- TRAs2_human[grep(x=TRAs2_human$max.tissue,"Panc"),]
PancreasTRAs3_human <- TRAs3_human[grep(x=TRAs3_human$max.tissue,"Panc"),]
PancreasTRAs4_mouse <- TRAs4_mouse[grep(x=TRAs4_mouse$max.tissue,"panc"),]
PancreasTRAs5_mouse <- TRAs5_mouse[grep(x=TRAs5_mouse$max.tissue,"panc"),]
PancreasTRAs6_human <- TRAs6_human[grep(x=TRAs6_human$max.tissue,"Panc"),]

# create character vectors with pancreas specific genes (for both humans & mice)
pancreas_gene_human <- unique(c(PancreasTRAs1_human$Symbol,PancreasTRAs2_human$gene.symbol,PancreasTRAs6_human$ensembl.symbol))
pancreas_gene_mouse <- unique(c(PancreasTRAs4_mouse$gene.symbol,PancreasTRAs5_mouse$gene.symbol))
TRA_pancreas_genes_human <- pancreas_gene_human
TRA_pancreas_genes_mouse <- pancreas_gene_mouse

length(pancreas_gene_human)
# 330
length(pancreas_gene_mouse)
# 59

# ----------------------------------------------------------
# all TRAs
# ----------------------------------------------------------
TRA_genes_human <- unique(c(TRAs1_human$Symbol,TRAs2_human$gene.symbol,TRAs6_human$ensembl.symbol))
TRA_genes_mouse <- unique(c(TRAs4_mouse$gene.symbol,TRAs5_mouse$gene.symbol))


```


```{r}
# ----------------------------------------------------------
# create pie charts for distribution of TRAs between tissues
# ----------------------------------------------------------

## labels of pie chart: tissues (percentage%)
label1 = paste0(rownames(GewebeTabelleProzent1), " (", as.character(round(GewebeTabelleProzent1[,"Anteil"],3)*100), "%)")
label2 = paste0(rownames(GewebeTabelleProzent2), " (", as.character(round(GewebeTabelleProzent2[,"Anteil"],3)*100), "%)")
label3 = paste0(rownames(GewebeTabelleProzent3), " (", as.character(round(GewebeTabelleProzent3[,"Anteil"],3)*100), "%)")
label4 = paste0(rownames(GewebeTabelleProzent4), " (", as.character(round(GewebeTabelleProzent4[,"Anteil"],3)*100), "%)")
label5 = paste0(rownames(GewebeTabelleProzent5), " (", as.character(round(GewebeTabelleProzent5[,"Anteil"],3)*100), "%)")
label6 = paste0(rownames(GewebeTabelleProzent6), " (", as.character(round(GewebeTabelleProzent6[,"Anteil"],3)*100), "%)")
## not all tissues included
label1[c(11:26,28:33)] = "" # pancreas is [27]
label2[c(11:15,17:61)] = ""
label3[c(11:64)] = "" # no pancreas mentioned here
label4[c(11:42,44:60)] = ""
label5[c(11:59,61:91)] = ""
label6[c(11:20,22:53)] = ""
```


```{r}
# TRAs1
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent1[,1], labels = label1, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in human tissues (Uhlen et al. 2015)")

setwd(paste(projectPath, "plots" ,sep = "/"))
#dev.copy2pdf(file = "TRA1_human_piechart.pdf")

# TRAs2
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent2[,1],labels = label2, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in human tissues (Su et al. 2002 & 2004)")

# TRAs3
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent3[,1],labels = label3, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in human tissues (Roth et al. 2008)")

# TRAs4 (mouse)
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent4[,1],labels = label4, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in mouse tissues")

setwd(paste(projectPath, "plots" ,sep = "/"))
#dev.copy2pdf(file = "TRA4_mouse_piechart.pdf")

# TRAs5 (mouse)
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent5[,1],labels = label5, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in mouse tissues")

# TRAs6
par(mai = c(0,0,0.5,0))
pie(GewebeTabelleProzent6[,1],labels = label6, col = colorRampPalette(brewer.pal(9,"Blues"))(9)[9:1], cex = 0.9, radius = 0.95, clockwise = TRUE, border = FALSE, main = "TRA expression in human tissues (GTEX 2013 & 2015)") 

remove(label1, label2, label3, label4, label5, label6) # not needed anymore
```

```{r}
# ----------------------------------------------------------
# pancreas specific TRAs distribution over chromosomes 
# ----------------------------------------------------------

# TRAs1 (human)

PancreasTRAs1ChromX <- gsub(x=PancreasTRAs1_human$Chromosome,pattern="X",replacement="23")
PancreasTRAs1ChromX <- as.integer(PancreasTRAs1ChromX)
PancreasTRAs1ChromXorder <- PancreasTRAs1ChromX[order(PancreasTRAs1ChromX)]

barplot(table(PancreasTRAs1ChromXorder),main="human pancreas specific TRAs (Uhlen et al. 2015)",xlab="chromosoms", ylab="frequency", col = rainbow(23), cex.names=.5, las=2, names.arg=c("1","2","3","5","6","7","8","9","10","11","12","14","15","16","17","18","19","20","22","X") )


# TRAs2 (human)

PancreasTRAs2ChromX <- gsub(x=PancreasTRAs2_human$chrom,pattern="X",replacement="23")
PancreasTRAs2ChromX <- as.integer(PancreasTRAs2ChromX)
PancreasTRAs2ChromXorder <- PancreasTRAs2ChromX[order(PancreasTRAs2ChromX)]

barplot(table(PancreasTRAs2ChromXorder),main="human pancreas specific TRAs (Su et al. 2002 & 2004)" ,xlab="chromosoms", ylab="frequency", col = rainbow(23), cex.names=.5, las=2, names.arg=c("1","2","3","4","5","6","7","9","10","11","12","13","14","15","16","17","19","20","22"))

# TRAs6 (human)

PancreasTRAs6ChromX <- gsub(x=PancreasTRAs6_human$ensembl.chrom,pattern="X",replacement="23")
PancreasTRAs6ChromX <- as.integer(PancreasTRAs6ChromX)
PancreasTRAs6ChromXorder <- PancreasTRAs6ChromX[order(PancreasTRAs6ChromX)]

barplot(table(PancreasTRAs6ChromXorder),main="human pancreas specific TRAs (GTEX 2013 & 2015)",xlab="chromosoms", ylab="frequency", col = rainbow(23), cex.names=.5, las=2, names.arg=c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X") )

# TRAs4 (mouse)

PancreasTRAs4ChromX <- gsub(x=PancreasTRAs4_mouse$chrom,pattern="X",replacement="20")
PancreasTRAs4ChromX <- as.integer(PancreasTRAs4ChromX)
PancreasTRAs4ChromXorder <- PancreasTRAs4ChromX[order(PancreasTRAs4ChromX)]

barplot(table(PancreasTRAs4ChromXorder),main="mouse pancreas specific TRAs4",xlab="chromosoms", ylab="frequency", col = rainbow(23), cex.names=.5)

# TRAs5 (mouse)

PancreasTRAs5ChromX <- gsub(x=PancreasTRAs5_mouse$chrom,pattern="X",replacement="20")
PancreasTRAs5ChromX <- as.integer(PancreasTRAs5ChromX)
PancreasTRAs5ChromXorder <- PancreasTRAs5ChromX[order(PancreasTRAs5ChromX)]

barplot(table(PancreasTRAs5ChromXorder),main="mouse pancreas specific TRAs5",xlab="chromosoms", ylab="frequency", col = rainbow(23), cex.names=.5, names.arg=c("1","2","3","4","5","6","7","8","9","10","12","15","17","19","X"))
```


## Gene expression  measured by microarrays
### Microarray Data sets

Microarrays were used to analyze the expression of a large number of genes in human pancreas cells. This was done by hybridizing labelled cDNA (target) to a specific DNA sequence (probe) with a defined position on the microarray chip. After hybridization the fluorescence emission was examined. \
The pancreas cell types used for this project are PDAC Capan-1 cells (GEO number GSE59761) and $\beta$-cells (GEO number GSE53454). These data sets were downloaded from the GEO database. For both the same Affymetrix chip was used (HG-U133 Plus 2) which means that the same transcripts were analyzed. \
Data set GSE59761 includes six chips. For three of those, the pancreatic cancer cells were transfected with unspecified small interfering RNA (siRNA) as negative control (NC). The other three were treated with siRNA targeted against the TBL1X gene, which encodes a transcriptional cofactor. This knockout causes decreased proliferation in the pancreatic cancer cells (Stoy \textit{et al.}, 2015). \
Data set GSE53454 consists of 24 chips. 13 of those form the control group, where the $\beta$-cells were observed over several time periods from 0 to 96 hours. The remaining eleven microarray chips provide gene expression data to $\beta$-cells that were treated with cytokines interleukin-1$\beta$ and interferon-$\gamma$ to simulate the actions of the immune system in diabetes type 1. This treatment was also conducted over a time period ranging from 1 to 96 hours (Lopes \textit{et al.}, 2014).


### Creating expression matrices

```{r}
# ----------------------------------------------------------
# create vsnrma normalized expression matrix
# ----------------------------------------------------------

set.seed(132) # with this vsnrma gives out the same values each time
# get CEL files
setwd(paste(projectPath, "rawdata", "GSE59761 PDAC Capan-1 Zellen", sep = "/"))
data_pancreas <- ReadAffy()
setwd(paste(projectPath, "rawdata", "GSE53454 Diabetes type 1", sep = "/"))
data_diabetes <- ReadAffy()
# change cdf
data_pancreas@cdfName <- "HGU133Plus2_Hs_ENST"
data_diabetes@cdfName <- "HGU133Plus2_Hs_ENST"
# create expression matrix
pancreas_matrix <- exprs(data_pancreas)
diabetes_matrix <- exprs(data_diabetes)
# store colnames (microarray)
pancreas_microarray_information <- colnames(pancreas_matrix)
diabetes_microarray_information <- colnames(diabetes_matrix)
# vsnrma normalization
pancreas_vsnrma <- vsnrma(data_pancreas)
diabetes_vsnrma <- vsnrma(data_diabetes)
# expression matrix of vsnrma normalized data set
pancreas_vsnrma_matrix <- exprs(pancreas_vsnrma)
diabetes_vsnrma_matrix <- exprs(diabetes_vsnrma)
# cut rownames at "." 
rownames(pancreas_vsnrma_matrix) <- str_replace(rownames(pancreas_vsnrma_matrix),"\\..*","") 
rownames(diabetes_vsnrma_matrix) <- str_replace(rownames(diabetes_vsnrma_matrix),"\\..*","")
# remove ending .CEL from colnames 
colnames(pancreas_vsnrma_matrix) <- str_remove(colnames(pancreas_vsnrma_matrix),"_.CEL")
colnames(diabetes_vsnrma_matrix) <- str_replace(colnames(diabetes_vsnrma_matrix),"\\..*","")
# remove control probes (AFFX) 
pancreas_vsnrma_matrix = pancreas_vsnrma_matrix[which(!startsWith(rownames(pancreas_vsnrma_matrix), "AFFX")),]
diabetes_vsnrma_matrix = diabetes_vsnrma_matrix[which(!startsWith(rownames(diabetes_vsnrma_matrix), "AFFX")),]
# transcript IDs
pancreas_transcript_names = rownames(pancreas_vsnrma_matrix)
diabetes_transcript_names = rownames(diabetes_vsnrma_matrix)

# ----------------------------------------------------------
# create data frames which hold information about micro arrays
# ----------------------------------------------------------

# create data frame with information about microarray samples

diabetes_microarrays = t(as.data.frame(
  sapply(diabetes_microarray_information, function(x){
    s = str_replace(x,"\\..*","")
    str_split(s, "_")
    })
  ))[,-c(2,3)]
colnames(diabetes_microarrays) <- c("number", "treatment", "time_h")
diabetes_microarrays <- transform(diabetes_microarrays, 
                                  time_h = as.integer(str_remove(diabetes_microarrays[,"time_h"],"h")))
rownames(diabetes_microarrays) <- c(1:24)
diabetes_microarrays <- as.data.frame(diabetes_microarrays)


pancreas_microarrays <- data.frame(number = substr(pancreas_microarray_information, 1,10), 
                                   version = substr(pancreas_microarray_information, 24,27),
                                   siRNA = c("TBL1", "TBL1", "TBL1", "NC", "NC", "NC"), 
                                   time = rep("24h",6),
                                   Affymetrix = rep("HG-U133_Plus_2",6))

# replace old IDs sample names
colnames(pancreas_matrix) <- pancreas_microarrays[,"number"]
colnames(diabetes_matrix) <- diabetes_microarrays[,"number"]
colnames(pancreas_vsnrma_matrix) <- pancreas_microarrays[,"number"]
colnames(diabetes_vsnrma_matrix) <- diabetes_microarrays[,"number"]

# ----------------------------------------------------------
# convert transcript IDs to gene symbols
# ----------------------------------------------------------

# get table for converting transcript IDs to gene symbols
setwd(paste(projectPath, "rawdata", "tables" ,sep = "/"))
annotation = read.csv("ensemble.103.txt") 
ensemble_transcripts  = annotation[,"Transcript.stable.ID"]
ensemble_symbol = annotation[,"HGNC.symbol"]
# name each gene symbol by their respective transcripts
names(ensemble_symbol) <- ensemble_transcripts
# select only those transcripts that are also present in the annotation
diabetes_GeneExprs <- diabetes_vsnrma_matrix[rownames(diabetes_vsnrma_matrix) %in% ensemble_transcripts,]
pancreas_GeneExprs <- pancreas_vsnrma_matrix[rownames(pancreas_vsnrma_matrix) %in% ensemble_transcripts,]
# create vector with the symbols of the transcript from expression matrix
diabetes_symbol <- ensemble_symbol[rownames(diabetes_GeneExprs)]
pancreas_symbol <- ensemble_symbol[rownames(pancreas_GeneExprs)]
# use gene symbols instead of transcripts as rownames
rownames(diabetes_GeneExprs) <- as.character(diabetes_symbol)
rownames(pancreas_GeneExprs) <- as.character(pancreas_symbol)
# order genes alphabetically & remove all those rows where "" is the rowname 
diabetes_GeneExprs <- diabetes_GeneExprs[order(rownames(diabetes_GeneExprs)),][1097:nrow(diabetes_GeneExprs),] 
pancreas_GeneExprs <- pancreas_GeneExprs[order(rownames(pancreas_GeneExprs)),][1097:nrow(pancreas_GeneExprs),] 
# replace colnames with the GSM number of the microarrays
colnames(diabetes_GeneExprs) <- diabetes_microarrays[,"number"]
colnames(pancreas_GeneExprs) <- pancreas_microarrays[,"number"]

# ----------------------------------------------------------
# pancreas specific TRAs
# ----------------------------------------------------------

# create a expression matrix with only those genes
diabetes_GeneExprs_sub = diabetes_GeneExprs[which(rownames(diabetes_GeneExprs) %in% TRA_pancreas_genes_human),]
pancreas_GeneExprs_sub = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% TRA_pancreas_genes_human),]

# ----------------------------------------------------------
# TRAs for all tissues
# ----------------------------------------------------------

pancreas_GeneExprs_allTRA = pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) %in% TRA_genes_human),]
diabetes_GeneExprs_allTRA = diabetes_GeneExprs[which(rownames(diabetes_GeneExprs) %in% TRA_genes_human),]

# ----------------------------------------------------------
# combining transcripts for same gene via median
# ----------------------------------------------------------

combineGeneExprs_median2 <- function(exprsmatrix){
  # create data frame with a column of all the gene symbols
  data <- data.frame(geneSymbols = rownames(exprsmatrix),
                     exprsmatrix, 
                     row.names = NULL)
  # group data frame by gene symbols and combine values of one gene by median
  combined <- aggregate(data[-1], by = list(data$geneSymbols) , FUN = median)
  # convert data frame to matrix (only expression values, no row names)
  result <- as.matrix(combined[,-1])
  # use gene symbols as row names
  rownames(result) <- combined[,1]
  # output
  result
}

pancreas_GeneExprs_combined <- combineGeneExprs_median2(pancreas_GeneExprs)
diabetes_GeneExprs_combined <- combineGeneExprs_median2(diabetes_GeneExprs)
pancreas_GeneExprs_sub_combined <- combineGeneExprs_median2(pancreas_GeneExprs_sub)
diabetes_GeneExprs_sub_combined <- combineGeneExprs_median2(diabetes_GeneExprs_sub)

```

The raw values from the microarrays are not suited to be used in the analysis. First, they have to be processed to allow comparison between the gene expression of different chips. An important part of this processing is the normalization using the 'vsnrma' function of the programming language R.

```{r, eval=FALSE, echo=TRUE, include=TRUE}
pancreas_vsnrma <- vsnrma(data_pancreas) # GSE59761 - PDAC data set
diabetes_vsnrma <- vsnrma(data_diabetes) # GSE53454 - beta-cell data set
```

This function does several steps at once: vsn normalization, background correction and summary of probesets. The resulting expression values are log 2 transformed.\
The resulting matrix contains the expression values for all transcripts for each chip. After translation of the transcripts into the corresponding gene symbols, pancreas-specific TRAs were selected. Then multiples of the same gene were combined by using the median of the respective expression values. Their expression values were included into the final expression matrix, which contains 250 out of the original 330 pancreas TRAs. 

```{r echo=FALSE, include=TRUE, eval=TRUE, collapse=TRUE}
kbl(pancreas_GeneExprs_sub_combined[1:5,], 
    caption = "Head of the final expression matrix of PDAC data set",
    booktabs = TRUE,
    align = "rrrrrr") %>%
  kable_styling(latex_options = c("striped", 
                                  "scale_down",
                                  "hold_position"),
                font_size = 10)

cat("Dimensions of the matrix:", dim(pancreas_GeneExprs_sub_combined))
```

The following graphs show the histogram, mean and median obtained from expression matrices before selecting the pancreas-specific TRAs. As seen in these graphs, both sets have a right-skewed distribution and are therefore not suitable for t-tests. Instead Wilcoxon rank-sum tests were used in the following statistical analysis. 

```{r}
mean(diabetes_GeneExprs_sub_combined)
median(diabetes_GeneExprs_sub_combined)
sd(diabetes_GeneExprs_sub_combined)


mean(pancreas_GeneExprs_sub_combined)
median(pancreas_GeneExprs_sub_combined)
sd(pancreas_GeneExprs_sub_combined)
```




```{r fig.cap = "Histogram of gene expression for both microarray data sets. The expression matrix used contains all genes from the microarray, where a gene is a summary of the corresponding transcripts.", fig.show="hold", fig.align="default", include=TRUE, eval=TRUE, out.width= "50%"}
# ----------------------------------------------------------
# distribution (all) gene expression: both data sets
# ----------------------------------------------------------

par(mai = c(0.85,0.85,0.5,0.1), cex.lab = 1.2, cex.main = 1.3)

hist(diabetes_GeneExprs_combined, 
     xlab = "gene expression",
     ylab = "probability",
     main = paste("Beta-cell data set"), 
     freq = FALSE, 
     xlim = c(4,16), 
     ylim = c(0,0.5)) 
abline(v = c(mean(diabetes_GeneExprs_combined), median(diabetes_GeneExprs_combined)), 
       col = c("red", "blue"), lwd = 2)
legend("topright", 
       legend = c(paste0("mean = ",round(mean(diabetes_GeneExprs_combined),2)), 
                  paste0("median = ", round(median(diabetes_GeneExprs_combined),2))), 
       fill = c("red", "blue"))

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "diabetes_histogram_microarrays.pdf")


par(mai = c(0.85,0.85,0.5,0.1))

hist(pancreas_GeneExprs_combined, 
     xlab = "gene expression",
     ylab = "probability",
     main = paste("PDAC data set"), 
     freq = FALSE, 
     xlim = c(4,16), 
     ylim = c(0,0.5))
abline(v = c(mean(pancreas_GeneExprs_combined), median(pancreas_GeneExprs_combined)), 
       col = c("red", "blue"), lwd = 2)
legend("topright", 
       legend = c(paste0("mean = ",round(mean(pancreas_GeneExprs_combined),2)), 
                  paste0("median = ", round(median(pancreas_GeneExprs_combined),2))), 
       fill = c("red", "blue"))

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_histogram_microarrays.pdf")
```


```{r eval=FALSE}
# ----------------------------------------------------------
# heatmap TRA gene expression with hclustering: diabetes data set
# ----------------------------------------------------------

anno <- select(diabetes_microarrays,treatment, time_h)
rownames(anno) <- diabetes_microarrays$number

pheatmap(diabetes_GeneExprs_sub, 
         labels_row = rep("",nrow(diabetes_GeneExprs_sub)),
         annotation_col = anno,
         treeheight_row = 0,
         main = "TRA gene expression in beta cells (GSE53454)")

# after combining
pheatmap(diabetes_GeneExprs_sub_combined, 
         labels_row = rep("",nrow(diabetes_GeneExprs_sub)),
         annotation_col = anno,
         treeheight_row = 0,
         main = "TRA gene expression in beta cells (GSE53454)\nafter combining transcripts from one gene over median")

# ----------------------------------------------------------
# heatmap gene expression with hclustering: PDAC data set
# ----------------------------------------------------------

# check if the knockout of TBL1X can be seen
anno <- select(pancreas_microarrays,siRNA)
rownames(anno) <- pancreas_microarrays$number

par(mai = c(1,0.1,2,1))
pheatmap(pancreas_GeneExprs[which(rownames(pancreas_GeneExprs) == "TBL1X"),], 
         main = "TBL1X - Gene expression in pancreatic cancer cells\ndata set: GSE59761 (Stoy et al., 2015)",
         annotation_col = anno,
         cluster_rows = TRUE, treeheight_row = 0, 
         fontsize_row = 8, fontsize_col = 7, fontsize = 9)

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_heatmap_TBL1X.pdf")

# pancreas TRAs
pheatmap(pancreas_GeneExprs_sub,
         annotation_col = anno,
         labels_row = rep("",nrow(pancreas_GeneExprs_sub)),
         cluster_rows = TRUE, treeheight_row = 0,
         fontsize_row = 8, fontsize_col = 7, fontsize = 9,
         main = "TRA gene expression in pancreatic cancer cells before combining via median\ndata set: GSE59761")

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_heatmap_TRAs.pdf")

# after combining
pheatmap(pancreas_GeneExprs_sub_combined,
         annotation_col = anno,
         labels_row = rep("",nrow(pancreas_GeneExprs_sub_combined)),
         cluster_rows = TRUE, treeheight_row = 0,
         fontsize_row = 8, fontsize_col = 7, fontsize = 9,
         main = "TRA gene expression in pancreatic cancer cells \n data set: GSE59761")

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_heatmap_TRA_combined.pdf")

# ------------------------------------------------
# boxplots: diabetes data set
# ------------------------------------------------

par(las = 2, 
    cex.axis = 0.5, cex.main = 0.8,
    mai = c(0.65,0.5,0.4,0.1), mfrow = c(2,1))

for (i in seq(1,250,by = 50)){
  
  generange = paste0("(", rownames(diabetes_GeneExprs_sub_combined)[i],
                     "-", 
                     rownames(diabetes_GeneExprs_sub_combined)[i+49],")")
  
  boxplot(t(diabetes_GeneExprs_sub_combined[i:(i+49),filter(diabetes_microarrays, treatment == "Control")$number]),
          col = rainbow(50), 
          main= paste0("boxplot of TRA gene expression in beta cells ", generange, "\ndata set: GSE53454"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  boxplot(t(diabetes_GeneExprs_sub_combined[i:(i+49),filter(diabetes_microarrays, treatment == "Cytok")$number]),
          col = rainbow(50),
          main= paste0("boxplot of TRA gene expression in beta cells treated with cytokines", generange, "\ndata set: GSE53454"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  filename = paste0("diabetes_geneexpression_boxplot_", 
                    rownames(diabetes_GeneExprs_sub_combined)[i],"_",
                    rownames(diabetes_GeneExprs_sub_combined)[i+49],".pdf")
  print(filename)
  setwd(paste(projectPath, "plots", sep = "/"))
  # dev.copy2pdf(file = filename)
}

# ------------------------------------------------
# boxplots: PDAC data set
# ------------------------------------------------

par(las = 2, 
    cex.axis = 0.5, cex.main = 0.8,
    mai = c(0.65,0.5,0.4,0.1), mfrow = c(2,1))

for (i in seq(1,250,by = 50)){
  
  generange = paste0("(", rownames(pancreas_GeneExprs_sub_combined)[i],
                     "-", 
                     rownames(pancreas_GeneExprs_sub_combined)[i+49],")")
  
  boxplot(t(pancreas_GeneExprs_sub_combined[i:(i+49),filter(pancreas_microarrays, siRNA == "NC")$number]),
          col = rainbow(50), 
          main= paste0("boxplot of TRA gene expression in pancreatic cancer cells ", generange, "\ndata set: GSE59761"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  boxplot(t(pancreas_GeneExprs_sub_combined[i:(i+49),filter(pancreas_microarrays, siRNA == "TBL1")$number]),
          col = rainbow(50),
          main= paste0("boxplot of TRA gene expression in pancreatic cancer cells after knockout of TBL1X ", generange, "\ndata set: GSE59761"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  filename = paste0("pancreas_geneexpression_boxplot_", 
                    rownames(diabetes_GeneExprs_sub_combined)[i],"_",
                    rownames(diabetes_GeneExprs_sub_combined)[i+49],".pdf")
  print(filename)
  setwd(paste(projectPath, "plots", sep = "/"))
  # dev.copy2pdf(file = filename)
}

```

\newpage
### Quality control of microarray chips

To ensure that the used microarrays are all of high quality and the potential mistakes made by the experimenter do not interfere with the analysis, a quality control was necessary. \
Looking at the images of each chip there was no abnormality detected for both data sets. Likewise, no considerable outliers were found between the microarrays after normalization. To determine this, boxplots of the microarrays and scatterplots were reviewed. The conclusion was to keep all chips from both data sets. 

\begin{table}[htb]
\resizebox{\textwidth}{!}{
\begin{tabular}{lllll}
\hline
affy 1.68.0 & hexbin 1.28.2 & ggplot2 3.3.3 & stats 4.0.3 & VennDiagram 1.6.20 \\ 
AnnotationDbi 1.52.0 & vsn 3.58.0 & ggfortify 0.4.11 & dplyr 1.0.5 & kableExtra 1.3.4 \\ 
hgu133plus2hsenstcdf 25.0.0  & rstudioapi 0.13 & ggrepel 0.9.1 &  reshape2 1.4.4 &  knitr 1.32 \\
hgu133plus2hsenstprobe 25.0.0  & tidyverse 1.3.1  & pheatmap 1.0.12 & factoextra 1.0.7 &  \\ 
\hline
\end{tabular}}
\caption{List of all packages used in R for obtaining, processing and analysis of data}
\end{table}

```{r eval=FALSE}
# ----------------------------------------------------------
# single chip control: diabetes data set
# ----------------------------------------------------------

setwd(paste(projectPath, "plots", sep = "/"))
par(pty = "s")

for (i in 1:24){
  image(data_diabetes[,i], col = rainbow(100, start = 0, end = 0.75)[100:1])
  file.name = paste0("diabetes_", str_remove(colnames(diabetes_matrix)[i],".CEL"),".pdf")
  # dev.copy2pdf(file = file.name)
}

# ----------------------------------------------------------
# single chip control: PDAC data set
# ----------------------------------------------------------

setwd(paste(projectPath, "plots", sep = "/"))
par(pty = "s")

for (i in 1:6){
  image(data_pancreas[,i], col = rainbow(100, start = 0, end = 0.75)[100:1])
  file.name <- paste0("pancreas_", str_remove(colnames(pancreas_matrix)[i],".CEL"),".pdf")
  #dev.copy2pdf(file = file.name)
}

# ----------------------------------------------------------
# RNA degradation plot: diabetes data set
# ----------------------------------------------------------

RNAdeg_diabetes = AffyRNAdeg(data_diabetes)

# use of shift & scale
par(pty = "s", mai = c(1.4,0.5,0.5,0.1))
plotAffyRNAdeg(RNAdeg_diabetes, col=rainbow(24), transform = "shift.scale")
title(sub = "beta-cell data set (GSE53454) rawdata")

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_rnadeg_rawdata.pdf")

# use only shift
par(pty = "s", mai = c(1.4,0.5,0.5,0.1))
plotAffyRNAdeg(RNAdeg_diabetes, col = rainbow(24), transform = "shift.only")
title(sub = "beta-cell data set (GSE53454) rawdata")

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_rnadegl_rawdata.pdf")

# ----------------------------------------------------------
# RNA degradation plot: PDAC data set
# ----------------------------------------------------------

RNAdeg_PDAC <- AffyRNAdeg(data_pancreas)

## shift and scale
par(pty = "s", mai = c(1.4,0.5,0.5,0.1))
plotAffyRNAdeg(RNAdeg_PDAC, col = rainbow(6))
title(sub = "PDAC data set (GSE59761) rawdata")

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "pancreas_rnadeg_shiftt_scale_rawdata.pdf")

## shift
par(pty = "s", mai = c(1.4,0.5,0.5,0.1))
plotAffyRNAdeg(RNAdeg_PDAC, col = rainbow(6), transform = "shift.only")
title(sub = "PDAC data set (GSE59761) rawdata")

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "pancreas_rnadeg_shift_rawdata.pdf")


# ----------------------------------------------------------
# meanSdPlot: diabetes data set
# ----------------------------------------------------------

meanSdPlot(diabetes_vsnrma, plot = FALSE)$gg + theme(aspect.ratio = 1)

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "diabetes_meanSdPlot_vsnrma_normalized.pdf")

# ----------------------------------------------------------
# meanSdPlot: PDAC data set
# ----------------------------------------------------------

meanSdPlot(pancreas_vsnrma, plot = FALSE)$gg + theme(aspect.ratio = 1) 

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_meanSdPlot_vsnrma_normalized.pdf")

# ----------------------------------------------------------
# histogram before and after normalization: diabetes data set
# ----------------------------------------------------------

par(mai = c(1.2,1,1,0.5))
hist(data_diabetes, 
     col = rainbow(24), 
     main = "Density function of log intensity of beta cells (data set: GSE53454)\nbefore normalization")
title(sub = "data from: (Lopes et al., 2014) ; (Marroqui et al., 2015)", cex.sub = 0.7, line = 4.5)

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_hist_rawdata.pdf")

par(mai = c(1.2,1,1,0.5))

plot(density(diabetes_vsnrma_matrix), 
     type = "n", xlab = "log Intensity", ylim = c(0,0.7), 
     main = "Density function of log intensity of beta cells (GSE53454) \n after normalization")
title(sub = "data from: (Lopes et al., 2014) ; (Marroqui et al., 2015)", cex.sub = 0.7, line = 4.5)

for (i in 1:ncol(diabetes_vsnrma_matrix)){
  lines(density(diabetes_vsnrma_matrix[,i]), col = rainbow(24)[i])
}

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_hist_vsnrma_normalized.pdf")

# ----------------------------------------------------------
# histogram before and after normalization: PDAC data set
# ----------------------------------------------------------

par(mai = c(1.2,1,1,0.5))
hist(data_pancreas, col = rainbow(6), main = "Density function of log intensity of PDAC cells (data set: GSE59761)\nbefore normalization")
title(sub = "data from: (Stoy et al., 2015)", cex.sub = 0.7, line = 4.5)

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_hist_rawdata.pdf")

par(mai = c(1.2,1,1,0.5))
plot(density(pancreas_vsnrma_matrix[,1]), type = "n", xlab = "log intensity", main = "Density function of log intensity of PDAC cells (data set: GSE59761)\nafter normalization", ylim = c(0,0.4))
title(sub = "data from: (Stoy et al., 2015)", cex.sub = 0.7, line = 4.5)

for (i in 1:ncol(pancreas_vsnrma_matrix)){
  lines(density(pancreas_vsnrma_matrix[,i]), col = rainbow(6)[i])
}

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "pancreas_hist_vsnrma_normalized.pdf")


# ----------------------------------------------------------
# boxplot before and after normalization: diabetes data set
# ----------------------------------------------------------

par(las = 2, mai = c(1.2,0.8,1,0.1))

boxplot(data_diabetes, names = substr(colnames(diabetes_matrix), 1,10),
        col = rainbow(24), cex.axis=0.6, ylab = "Intensity values", 
        main="Gene expression in mircoarrays (beta cell data set: GSE53454)\nbefore vsnrma normalization", 
        horizontal = FALSE)
title(sub = "data from: (Lopes et al., 2014) ; (Marroqui et al., 2015)", cex.sub = 0.7, line = 4.5)

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_boxplot_rawdata.pdf")

par(las = 2, mai = c(1.2,0.8,1,0.1))

boxplot(diabetes_vsnrma_matrix, names = substr(colnames(diabetes_matrix), 1,10),
        col = rainbow(24), cex.axis = 0.6, ylab = "Intensity values",
        main= "Gene expression in mircoarrays (beta cell data set: GSE53454)\nafter vsnrma normalization", 
        horizontal = FALSE, outline = FALSE) # outliers are not drawn
title(sub = "data from: (Lopes et al., 2014) ; (Marroqui et al., 2015)", cex.sub = 0.7, line = 4.5)
abline(v = median(diabetes_vsnrma_matrix))

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "diabetes_boxplot_vsnrma_normalized.pdf") 


# ----------------------------------------------------------
# boxplot before and after normalization: PDAC data set
# ----------------------------------------------------------

par(las = 2, mai = c(1.2,0.8,1,0.1))

boxplot(data_pancreas, names = pancreas_microarrays$number, col = rainbow(6), main = "Gene expression in mircoarrays (PDAC data set: GSE59761)\nbefore normalization", cex.axis=0.6, ylab = "Intensity values")
title(sub = "data from: (Stoy et al., 2015)", cex.sub = 0.7, line = 4.5)

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "pancreas_boxplot_rawdata.pdf")

par(las = 2, mai = c(1.2,0.8,1,0.1))

boxplot(data_pancreas, names = pancreas_microarrays$number, col = rainbow(6), main = "Gene expression in mircoarrays (PDAC data set: GSE59761)\nafter normalization", cex.axis=0.6, ylab = "Intensity values")
title(sub = "data from: (Stoy et al., 2015)", cex.sub = 0.7, line = 4.5)

setwd(paste(projectPath, "plots" ,sep = "/"))
# dev.copy2pdf(file = "pancreas_boxplot_vsnrma_normalized.pdf")
# ----------------------------------------------------------
# scatterplots: diabetes data set
# ----------------------------------------------------------

setwd(paste(projectPath, "plots", "scatter plots diabetes" ,sep = "/"))
par(mai = c(0.8,0.8,0.6,0.1), pty = "s")

for (i in 1:23){
  for (j in (i+1):24){
    plot(diabetes_vsnrma_matrix[,i], diabetes_vsnrma_matrix[,j], pch = ".",
         xlab = substr(colnames(diabetes_matrix), 1,10)[i],
         ylab = substr(colnames(diabetes_matrix), 1,10)[j])
    abline(0,1,col="red")

    title(main = paste("Scatterplot of microarrays (GSE53454)\n", 
                       substr(colnames(diabetes_matrix), 1,10)[i],
                       "and",
                       substr(colnames(diabetes_matrix), 1,10)[j],
                       sep = " ", collapse = NULL))
    
    file.name = paste0("diabetes_scatterplot_",
                       substr(colnames(diabetes_matrix), 1,10)[i],
                       ".pdf")
    # dev.copy2pdf(file = file.name)
  }
}

# ----------------------------------------------------------
# scatterplots: PDAC data set
# ----------------------------------------------------------

setwd(paste(projectPath, "plots" ,sep = "/"))
par(mai = c(0.8,0.8,0.6,0.1), pty = "s")

for (i in 1:5) {
  for (j in (i+1):6){
    plot(pancreas_vsnrma_matrix[,i], pancreas_vsnrma_matrix[,j], pch = ".",
         xlab = pancreas_microarrays$number[i],
         ylab = pancreas_microarrays$number[j])
    abline(0, 1, col = "red")
  
    title(main = paste("Scatterplot of microarrays\n", 
                       pancreas_microarrays$number[i],
                       "and",
                       pancreas_microarrays$number[j],
                       sep = " ", collapse = NULL))
    
    file.name = paste("pancreas_scatterplot_",
                      pancreas_microarrays$number[i], 
                      "_",  
                      pancreas_microarrays$number[j], 
                      ".pdf", 
                      sep = "")
  
  # dev.copy2pdf(file = file.name)
  }}

```


# Analysis

##  Identification of a possible biomarker or therapeutic target for pancreatic cancer

### Introduction 

The high mortality of pancreatic cancer is partly due to the cancer commonly only being diagnosed at advanced stages, where the treatment options are limited including surgical resection due to metastasis. Another reason are asymptomatic or unspecific symptoms, further complicated by the absence of specific diagnostic and predictive biomarkers for early diagnosis and treatments(McGuigan \textit{et al.},2018). \
The most widely used biomarker is Serum carbohydrate antigen CA 19-9 but because of low sensitivity and specificity (both around 80%) it is not used as a universal screening tool. Rather CA 19-9 is utilized for high risk patients as well as for detection of recurrence and assessment of response to treatment (Hongbeom et al, 2020). \
In brief, this shows why it is so important to extend the search for other biomarkers in hope of improving both the diagnostic as well as the treatment options. \
In this first part of the report, the focus is on the two control sets and determining if highly expressed pancreas specific antigens in the cancer cells but not in the neighboring $\beta$-cells can be found.

###  Identifying differentially expressed genes

```{r}
# ------------------------------------------------
# genes that are highly expressed in beta cells (over certain threshold)
# ------------------------------------------------

# beta cells are the control group from diabetes data set
diabetes_GeneExprs_sub_control <- diabetes_GeneExprs_sub_combined[, filter(diabetes_microarrays,treatment == "Control")$number]
diabetes_GeneExprs_sub_treat <- diabetes_GeneExprs_sub_combined[, filter(diabetes_microarrays,treatment == "Cytok")$number]

# Median of pancreas TRAs in control group
diabetes_ctrl_median <- apply(diabetes_GeneExprs_sub_control, 1, median, na.rm=TRUE)

median(diabetes_GeneExprs_sub_control)
## 8.083549

threshold = 1.8 * median(diabetes_GeneExprs_sub_control)

## Search for genes, whose median gene expression are more than threshold times that of the median of control group.
names(which(diabetes_ctrl_median > threshold))
## Names of genes that fit this criteria: 2 in total.
## INS, PRSS2

## Expression values (control & treatment groups) of those genes:
diabetes_abovethreshold <- diabetes_GeneExprs_sub_combined[names(which(diabetes_ctrl_median > threshold)),]
diabetes_abovethreshold

remove(threshold) # not needed anymore

# ------------------------------------------------
# Plots: change over time
# ------------------------------------------------

par(mfrow = c(2,2), mai = c(0.75,0.7,0.5,0.1))

## control

for (i in 1:nrow(diabetes_abovethreshold)){
  plot(diabetes_abovethreshold[i, arrange(filter(diabetes_microarrays,treatment == "Control"), time_h)$number],
       type = "b", xaxt = "n",
       xlab = "time [h]", ylab = "Gene expression",
       main = paste(rownames(diabetes_abovethreshold)[i], "\n", "Control"))
  axis(1, at = c(1:ncol(diabetes_GeneExprs_sub_control)), cex.axis = 0.8,
       labels = arrange(filter(diabetes_microarrays,treatment == "Control"), time_h)$time_h)
}

## treatment

for (i in 1:nrow(diabetes_abovethreshold)){
  plot(diabetes_abovethreshold[i, filter(diabetes_microarrays,treatment == "Cytok")$number],
       type = "b", xaxt = "n",
       xlab = "time [h]", ylab = "Gene expression",
       main = paste(rownames(diabetes_abovethreshold)[i], "\n", "Treatment"))
  axis(1, at = c(1:ncol(diabetes_GeneExprs_sub_treat)), cex.axis = 0.8,
       labels = arrange(filter(diabetes_microarrays,treatment == "Cytok"), time_h)$time_h)
}

```

```{r}
# ------------------------------------------------
# genes that are highly expressed in PDAC cells (over certain threshold)
# ------------------------------------------------
# normal pancreatic cancer cells are in control group
pancreas_GeneExprs_sub_control <- pancreas_GeneExprs_sub_combined[, filter(pancreas_microarrays, siRNA == "NC")$number ]
pancreas_GeneExprs_sub_treat <- pancreas_GeneExprs_sub_combined[ , filter(pancreas_microarrays, siRNA == "TBL1")$number]


# Median of pancreas TRAs in control group
pancreas_ctrl_median <- apply(pancreas_GeneExprs_sub_control,1,median)

median(pancreas_GeneExprs_sub_control)
## 6.771193

threshold <- 1.8 * median(pancreas_GeneExprs_sub_control)
threshold
## 12.18815

## Search for genes, whose median gene expression are more than threshold times that of the median of control group.
pancreas_abovethreshold <- pancreas_GeneExprs_sub_combined[names(which(pancreas_ctrl_median > threshold)),]
pancreas_abovethreshold


# ------------------------------------------------
# Plots: highly expressed genes in PDAC cells
# ------------------------------------------------

par(mai = c(0.8,0.8,0.6,0.1))

boxplot(t(pancreas_GeneExprs_sub_control[(which(pancreas_ctrl_median > threshold )),]),col = rainbow(8), main = paste0("Highly expressed TRA genes in the PDAC dataset (GSE59761)","\n","higher than ",round(threshold,2)),ylab="gene expression value",xlab="Genes")

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "pancreas_boxplot_highexpression.pdf")

remove(threshold)

```


```{r}
#Creating a empty vector for the p-values
Differential_TRA_values <- vector("numeric",length(diabetes_GeneExprs_sub_control[,1]))
Differential_TRA_values2 <- vector("numeric",length(diabetes_GeneExprs_sub_control[,1]))

# ------------------------------------------------
# Higher differential expressed genes
# ------------------------------------------------

# for-loop to calculate the p-values for each gene 
# selecting genes with p-value < 0.05
for (i in 1:length(diabetes_GeneExprs_sub_control[,1])){
  WilcoxonTest <- wilcox.test(x=pancreas_GeneExprs_sub_control[i,],y=diabetes_GeneExprs_sub_control[i,],alternative ="greater")
  if (WilcoxonTest$p.value < 0.05){
    Differential_TRA_values[i] <- WilcoxonTest$p.value}
    
}
# Creating a matrix for higher expressed PDAC genes with all the gene symbols and their corresponding p-values
Gensymbol_pvalue_higher_matrix <- matrix(c(rownames(pancreas_GeneExprs_sub_control),Differential_TRA_values),nrow=length(diabetes_GeneExprs_sub_control[,1]),byrow=FALSE)
colnames(Gensymbol_pvalue_higher_matrix) <- c("Genesymbol","p-value")


# Removing all rows with a p-value = 0 (the p-value was bigger than 5%)
Gensymbol_pvalue_higher_matrix <- Gensymbol_pvalue_higher_matrix[-which(Gensymbol_pvalue_higher_matrix[,2]==0),]
dim(Gensymbol_pvalue_higher_matrix)
## 40  2 

# ------------------------------------------------
# lower  differential expressed genes
# ------------------------------------------------

# for-loop to calculate the p-values for each gene 
# selecting genes with p-value < 0.05
for (i in 1:length(diabetes_GeneExprs_sub_control[,1])){
  WilcoxonTest2 <- wilcox.test(x=pancreas_GeneExprs_sub_control[i,],y=diabetes_GeneExprs_sub_control[i,],alternative ="less")
  if (WilcoxonTest2$p.value < 0.05){
    Differential_TRA_values2[i] <- WilcoxonTest2$p.value}
}

# Creating a matrix for lower expressed PDAC genes with all the gene symbols and their corresponding p-values
Gensymbol_pvalue_lower_matrix <- matrix(c(rownames(pancreas_GeneExprs_sub_control),Differential_TRA_values2),nrow=length(diabetes_GeneExprs_sub_control[,1]),byrow=FALSE)
colnames(Gensymbol_pvalue_lower_matrix) <- c("Genesymbol","p-value")

# Removing all rows with a p-value = 0 (the p-value was bigger than 5%)
Gensymbol_pvalue_lower_matrix <- Gensymbol_pvalue_lower_matrix[-which(Gensymbol_pvalue_lower_matrix[,2]==0),]
dim(Gensymbol_pvalue_lower_matrix)
## 190   2

## 40 genes are higher expressed in pancreatic cancer cells in comparison to ß cells 
## 190 genes are lower expressed in pancreatic cancer cells in comparison to ß cells
## 20 genes show no significant difference between the two data sets 

# ------------------------------------------------
# Selecting differential expressed genes
# ------------------------------------------------

#Creating a empty vector for the p-values

Differential_TRA_values3 <- vector("numeric",length(diabetes_GeneExprs_sub_control[,1]))

# for-loop to calculate the p-values for each gene 
# selecting genes with p-value < 0.05

for (i in 1:length(diabetes_GeneExprs_sub_control[,1])){
  WilcoxonTest3 <- wilcox.test(x=diabetes_GeneExprs_sub_control[i,],y=pancreas_GeneExprs_sub_control[i,],alternative ="two.sided")
  if (WilcoxonTest3$p.value < 0.05){
    Differential_TRA_values3[i] <- WilcoxonTest3$p.value}
    
}

# Creating a matrix with all the gene symbols and their corresponding p-values

Gensymbol_pvalue_differential_matrix <- matrix(c(rownames(pancreas_GeneExprs_sub_control),Differential_TRA_values3),nrow=length(diabetes_GeneExprs_sub_control[,1]),byrow=FALSE)
colnames(Gensymbol_pvalue_differential_matrix) <- c("Genesymbol","p-value")

# Removing all rows with a p-value = 0 (the p-value was bigger than 5%)

Gensymbol_pvalue_differential_matrix  <- Gensymbol_pvalue_differential_matrix[-which(Gensymbol_pvalue_differential_matrix[,2]==0),]

dim(Gensymbol_pvalue_differential_matrix)
## 225   2

## 225 genes are differential expressed with p = 5%

# ------------------------------------------------
# Creating matrix p-value + Mean Difference
# ------------------------------------------------

#Creating matrix that only contains differential expressed genes and their corresponding median difference between the two control sets


# Calculating the median values for each of the genes
diabetescontrolmedian <- apply(diabetes_GeneExprs_sub_control,1,median)
pancreascontrolmedian <- apply(pancreas_GeneExprs_sub_control,1,median)

# Calculating the median difference
differencemedian <-  pancreascontrolmedian - diabetescontrolmedian
differencemedianabs <- abs(pancreascontrolmedian - diabetescontrolmedian)

# Creating matrix with the median values for each of the genes
differencemedianmatrix <-  data.frame(rownames(pancreas_GeneExprs_sub_control),differencemedian,diabetescontrolmedian,pancreascontrolmedian)
differencemedianmatrixabs <-  data.frame(rownames(pancreas_GeneExprs_sub_control),differencemedianabs,diabetescontrolmedian,pancreascontrolmedian)

# Gene names of diferentially expressed genes
differentialexpressedgenenames <- Gensymbol_pvalue_differential_matrix[,1]

# Selecting only the genes that are differential expressed
differentialdifferencematrix <- differencemedianmatrix[which((rownames(diabetes_GeneExprs_sub_control) %in% differentialexpressedgenenames)),]
differentialdifferencematrixabs <- differencemedianmatrixabs[which((rownames(diabetes_GeneExprs_sub_control) %in% differentialexpressedgenenames)),]

# Matrix in descending order

differentialdifferencematrixordered <- differentialdifferencematrix[order(differentialdifferencematrix[,2],decreasing = TRUE),]
differentialdifferencematrixorderedabs <- differentialdifferencematrixabs[order(differentialdifferencematrixabs[,2],decreasing = TRUE),]

colnames(differentialdifferencematrixordered) <- c("Gensymbol","Median Difference","Median Diabetes","Median Pancreas")
colnames(differentialdifferencematrixorderedabs) <- c("Gensymbol","Median Difference","Median Diabetes","Median Pancreas")

```



```{r}
# ------------------------------------------------
# log2foldchange
# ------------------------------------------------

log2.fold.change <- function(group1, group2){
  
  # only if both groups have the same row numbers
  if(nrow(group1) == nrow(group2)){
    
    # compute log2(group1/group2) of median values 
    # values are already log transformed!
    log2_foldchange = sapply(1:nrow(group1), function(x){
      g1_m = median(as.numeric(group1[x,]))
      g2_m = median(as.numeric(group2[x,]))
      g1_m - g2_m # log2(group1/group2) 
    })
    
    # wilcoxon rank sum test group1 vs group2 for each gene 
    w = sapply(1:nrow(group1), function(x){
      g1 = as.numeric(group1[x,])
      g2 = as.numeric(group2[x,])
      wilcox.test(g1, g2, exact = FALSE)$p.value
    })
    
    trend_w = 1:length(w)
    trend_w[w < 0.05 & log2_foldchange > 0] = "up" # group1 higher 
    trend_w[w < 0.05 & log2_foldchange < 0] = "down" # group2 higher
    trend_w[w >= 0.05] = "not significant" 
  
    # output
    data.frame(GeneSymbol = rownames(group1), # genes
               log2_foldchange = log2_foldchange,  # log2 foldchange
               wilcox_raw_pvalue = w, # wilcoxon rank sum test
               wilcox_log10_pvalue = -1*log10(w),
               wilcox_trend = trend_w)
    
  }else{print("Something has gone wrong!")}
}

differential_control_diabetes_pancreas_fc = log2.fold.change(pancreas_GeneExprs_sub_control, diabetes_GeneExprs_sub_control) # ratio = control PDAC cells / beta cells

arrange(differential_control_diabetes_pancreas_fc, -log2_foldchange) # sort by how much differential expressed
filter(differential_control_diabetes_pancreas_fc, wilcox_trend != "not significant") # only significant genes
```

```{r}
# ------------------------------------------------
# highly and differential expressed genes in pancreatic cancer
# ------------------------------------------------

threshold <- 1.7*median(pancreas_GeneExprs_sub_control)
nameshigherpanc <- rownames(pancreas_GeneExprs_sub_control[which(pancreas_ctrl_median > threshold),])

# Are the highly expressed genes part of the vector with differential expressed genes

differentialdifferencematrixordered[which(differentialdifferencematrixordered[,1] %in% nameshigherpanc),]

filter(differential_control_diabetes_pancreas_fc, GeneSymbol %in% nameshigherpanc, wilcox_trend == "up")

match(c("ANXA4","STT3A","TSPAN8"), arrange(differential_control_diabetes_pancreas_fc, -log2_foldchange)$GeneSymbol) # how much differential are these?
## TSPAN - most differential in up regulation
## STT3A - 4. place
## ANXA4 - 18. place


filter(arrange(differential_control_diabetes_pancreas_fc, -log2_foldchange), wilcox_trend != "not significant")
differentialdifferencematrixordered

filter(differential_control_diabetes_pancreas_fc, GeneSymbol %in% nameshigherpanc)

# for threshold = 1.8*median:
## TSPAN8 and GNAS are both highly expressed PDAC Genes and differential expressed
## Problem: GNAS in diabetes higher than pancreas 
## RPL5,RPL8,RPS9 are Ribosomal genes and not differential expressed

```

The approach here was to first identify differentially expressed genes between the untreated $\beta$- and PDAC cells. For this a Wilcoxon rank-sum test was performed for each gene.

```{r eval=FALSE, echo=TRUE, include=TRUE}
# for-loop to calculate the p-values for each gene 
for (i in 1:length(diabetes_GeneExprs_sub_control[,1])){
  # beta cell vs pancreatic cancer cell (PDACs) gene expression
  WilcoxonTest3 <- wilcox.test(x = diabetes_GeneExprs_sub_control[i,], 
                               y = pancreas_GeneExprs_sub_control[i,], 
                               alternative ="two.sided") 
  # selecting genes with p-value < 0.05
  if (WilcoxonTest3$p.value < 0.05){
    Differential_TRA_values3[i] <- WilcoxonTest3$p.value}}
```

It determined whether the difference in gene expression was significant or due to statistical fluctuation. \
With the p-value set at p < 0.05, 225 out of the 250 genes are significantly differentially expressed. On the basis of those genes a matrix was compiled with the median gene expression of both data sets and also the difference between them to quantify the genes. 

```{r eval=TRUE, include=TRUE}

kbl(head(differentialdifferencematrixordered[1:3,-1]), 
    caption = "Head of the table of the matrix with the median value and median difference of the differentially expressed genes",
    booktabs = TRUE,
    align = "rrr") %>%
  kable_styling(latex_options = c("striped",
                                  "hold_position"),
                font_size = 10)

```

In this context  the log2 fold change was calculated and a volcano plot was created with that information. It shows the significance versus the magnitude, quantifying the difference in gene expression of a gene.

```{r fig.cap = "Genes that are significantly higher expressed in the PDAC cells are represented in orange. A higher expression in beta-cells leads to the blue color. Genes with a p-value higher than 0.05 in the Wilcoxon Rank sum test are marked as not significant in grey.", include=TRUE, eval=TRUE}

theme_set(theme_bw())
ggplot(data = differential_control_diabetes_pancreas_fc, aes(x = log2_foldchange, y = wilcox_log10_pvalue, label = GeneSymbol, col = wilcox_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") +
  labs(title = "Volcano plot (Wilcox test) for control vs. control (GSE53454 & GSE59761)", 
       x = "log2 fold change", y = "-log10 (p-value)", colour = "trend",
       subtitle = "ratio = control PDAC cells / beta cells") +
  xlim(c(-max(abs(differential_control_diabetes_pancreas_fc$log2_foldchange)),
         max(abs(differential_control_diabetes_pancreas_fc$log2_foldchange))))+ 
  scale_color_manual(values = c("navyblue", "grey", "orange"), 
                     labels = c("higher in beta-cells", "not significant", "higher in PDAC cells")) + 
  geom_text_repel(size = 2, max.overlaps = 35, show.legend = FALSE) +
  theme(legend.title = element_text(size = 14),
        legend.text = element_text(size = 12))

setwd(paste(projectPath, "plots", sep = "/"))
# dev.copy2pdf(file = "controlcontrol_vulcano_wilcox.pdf")
```


```{r}
###Venn-Diagramm (highly expressed ẞ-cells, highly expressed PDAC, differently expressed)

threshold <- 1.7*median(diabetes_GeneExprs_sub_treat)
nameshigherdiabetes <- rownames(diabetes_GeneExprs_sub_treat[which(diabetes_ctrl_median > threshold),])

x <- list(
  A = nameshigherpanc, 
  B = nameshigherdiabetes, 
  C = differentialdifferencematrixordered[,1]
  
  )


display_venn <- function(x, ...){
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}
display_venn(x, category.names = c("high in PDAC" , "high in treated ß-cells " , "differently expressed"),
# Circles
        lwd = 2,
        lty = 'blank',
        fill = c("deepskyblue", "deeppink", "yellow"),
        # Numbers
        cex = .9,
        fontface = "italic",
        # Set names
        cat.cex = 1,
        cat.fontface = "bold",
        cat.default.pos = "outer",
      cat.dist = c(-0.05, -0.25 , -0.05)
)
```
### Selecting possible biomarkers 

Now having identified the differential expressed genes - for a possible biomarker they also would have to be highly expressed in the cancer cells. For this the gene expression of the differential expressed genes was compared against the highest expressed genes in the cancer cell. \
These genes were than checked against existing literature in context of an association to (pancreatic) cancer. Out of all those genes two stood out - several publications indicate TSPAN8 and GMNN as possible biomarkers for pancreatic cancer. \
TSPAN8 is a member of the tetraspanin family that builds tetraspanin-enriched microdomains which are important in the signal transduction. The TSPAN8-mediated protein complexes interact with themselves and various other cellular signalling molecules.
During their normal physiological function they are involved in the regulation of multiple processes like cell adhesion, proliferation and differentiation. But several publications have linked the overexpression of TSPAN8 to the progression and metastasis of several cancers by interacting with various binding partners (Kyun Heo, Sumook Lee ,2020). \
To note is also that tetraspanins are constitutive exosome components but a higher TSPAN8 expression could be found on exosomes. This could lead to a possible non-invasive cancer diagnostic – a blood serum test (Wang et al.,2013). \
Geminin (GMNN) is responsible for the regulation of DNA replication in interaction with Cdc10-dependent transcript (Cdt1). Overexpression of geminin leads to replication defects and genomic instability, which are associated with cancer formation (Kushwaha \textit{et al.}, 2016). An unrelated study confirmed that geminin is overexpressed in pancreatic cancer (Salabat \textit{et al.}, 2008).

## Fate of pancreatic cells determined by TRA expression

### Difference in gene expression in diabetes data set

$\beta$-cells have a highly increased cell death rate in diabetes type 1 (Eizirik \textit{et al.}, 2020), while pancreatic cancer cells are known for their fast proliferation (Takakura \textit{et al.}, 2015). Here the gene expression of TRAs in treated and untreated $\beta$- and pancreatic cancer cells is compared to establish similarities and differences in their intracellular pathways leading them to their respective fates. \
First, the difference in healthy and stressed $\beta$-cells was deduced to determine which genes are influenced by the cytokine treatment. \
To achieve this goal, a Principal Component Analysis (PCA) was conducted, which can be seen below. The optimal number of two clusters was determined by the silhouette method. Those clusters for the microarray chips were allocated by k-means. The conclusion was reached that the pancreatic-specific TRA expression overall is not considerably affected by the cytokine treatment but rather by the temporal component.

```{r}
# ------------------------------------------------
# PCA & kmeans
# ------------------------------------------------

# create a data frame (columns = treatment, time_h, gene expression)
diabetes_df <- cbind(treatment = diabetes_microarrays[,"treatment"], 
            time_h = diabetes_microarrays[,"time_h"],
            data.frame(t(diabetes_GeneExprs_sub_combined)))

# PCA on the TRA expressions (scaled & centered)
diabetes_pca <- prcomp(diabetes_df[,-c(1,2)], scale. = TRUE, center = TRUE)

# elbow method 
set.seed(123)
fviz_nbclust(diabetes_pca$x, kmeans, method = "wss")
## not clear answer (no kink)

#silhouette method plot
fviz_nbclust(diabetes_pca$x, kmeans, method = "silhouette")
## cluster number = 2 is highest 

```

```{r fig.cap = "The Principal Component Analysis of the beta-cell data set, in which the microarray chips were divided by their specific treatment as well as the temporal component. Dots in red represent chips of the control group and therefore untreated beta-cells, while the color blue represents beta-cells treated with cytokines. The size of the dots increases with time that has passed.", include=TRUE, eval=TRUE}

# plot PC1 against PC2 
theme_set(theme_bw())
autoplot(diabetes_pca, data = diabetes_df, 
         colour = "treatment", size = "time_h", 
         main = "PCA of beta-cell data set (GSE53454)",
         xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1) + 
  scale_color_manual(values = c("firebrick1", "cornflowerblue"),
                     labels = c("Control", "Cytokine")) + 
  theme(legend.title = element_text(size = 14),
        legend.text = element_text(size = 12))

```

```{r}

# plot kmeans for k = 2 
autoplot(kmeans(diabetes_pca$x,2), data = diabetes_pca$x, frame = TRUE, xlim = c(-0.5,0.5), ylim = c(-0.5,0.5), asp = 1, main = "kmeans clustering (k=2) on beta-cell data set (GSE53454)") + 
  scale_color_brewer(palette = "Set1") + 
  scale_fill_brewer(palette = "Set1")

# add cluster numbers to already existing data frame (rows = microarrays)
diabetes_pca_kmeans2 <- kmeans(diabetes_pca$x,2)
diabetes_pca_kmeans4 <- kmeans(diabetes_pca$x,4)
diabetes_df <- cbind(kmeans2 = diabetes_pca_kmeans2$cluster, kmeans4 = diabetes_pca_kmeans4$cluster, diabetes_df)
(diabetes_df)
sort(filter(diabetes_df,kmeans2 == 1)$time_h) # higher in general
sort(filter(diabetes_df,kmeans2 == 2)$time_h) # lower in general
```

Moreover, a volcano plot was created reflecting the gene expression of treated versus untreated $\beta$-cells. A Wilcoxon rank-sum test was used to extract single genes that show a significant difference in their gene expression before and after cytokine treatment.\
A significantly upregulated gene in the treated group is CXCL11. This gene is known to be induced by IFN-$\gamma$ (Tokunaga \textit{et al.}, 2018), which was part of the treatment. This gene induces leukocytic infiltration into pancreatic tissue, that leads to $\beta$-cell destruction and ultimately, to type 1 diabetes mellitus (Burke \textit{et al.}, 2016).

```{r}
# ------------------------------------------------
# log2foldchange treatment vs control 
# which genes are influenced by cytokine treatment
# ------------------------------------------------

diabetes_fc_treatment = log2.fold.change(diabetes_GeneExprs_sub_treat, diabetes_GeneExprs_sub_control)
# ratio = treatment / control 

ggplot(data = diabetes_fc_treatment, aes(x = log2_foldchange, y = wilcox_log10_pvalue, label = GeneSymbol, col = wilcox_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") +
  labs(title = "Volcano plot (Wilcox test) for control vs. treatment \nin beta-cell data set (GSE53454)",
       subtitle = "ratio = treatment / control", 
       x = "log2 fold change", y = "-log10 (p-value)",
       colour = "trend") +
  xlim(c(-max(abs(diabetes_fc_treatment$log2_foldchange)),
         max(abs(diabetes_fc_treatment$log2_foldchange)))) + 
  scale_color_manual(values = c("navyblue", "grey", "orange"),
                     labels = c("higher in control group", "not significant", "higher in treatment group")) + 
  geom_text_repel(size = 2, max.time = 50, show.legend = FALSE)

```

### Difference in gene expression in PDAC data set

The same steps were followed for the PDAC cells as for the analysis of the $\beta$-cells. This time the k-means following the PCA detected two clusters: control and treatment. There was a definite divide in the overall TRA expression between these groups though no single differing gene could be identified through Wilcoxon rank-sum test.

```{r}
pancreas_df <-cbind(treatment = pancreas_microarrays[,"siRNA"],
            data.frame(t(pancreas_GeneExprs_sub_combined)))


# PCA on the TRA expressions
pancreas_pca <- prcomp(pancreas_df[,-1], scale. = TRUE)

# plot PC1 against PC2
autoplot(pancreas_pca, data = pancreas_df, colour = "treatment", main = "Principal Component Analysis on PDAC data set (GSE59761)", xlim = c(-0.8,0.8), ylim = c(-0.8,0.8), asp = 1)

# plot kmeans for k = 2
autoplot(kmeans(pancreas_pca$x,2), data = pancreas_pca$x, frame = TRUE, xlim = c(-0.8,0.8), ylim = c(-0.8,0.8), asp = 1, main = "kmeans clustering (k=2)\nafter PCA on PDAC data set (GSE59761)") + 
  scale_color_hue(direction = -1, h.start = 90) +
  scale_fill_hue(direction = -1, h.start = 90)

# add cluster numbers to already existing data frame (rows = microarrays)
pancreas_df <- cbind(kmeans2 = kmeans(pancreas_pca$x,2)$cluster, pancreas_df)
head(pancreas_df)

# ------------------------------------------------
# log2foldchange treatment vs control 
# which genes are influenced by gene knockout
# ------------------------------------------------


pancreas_fc = log2.fold.change(pancreas_GeneExprs_sub_treat, pancreas_GeneExprs_sub_control)
# ratio = treatment/control
pancreas_fc


ggplot(data = pancreas_fc, aes(x = log2_foldchange, y = wilcox_log10_pvalue, label = GeneSymbol, col = wilcox_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") + 
  labs(title = "Volcano plot (Wilcox test) for PDAC data set (GSE59761)", 
       x = "log2 fold change", y = "-log10 (p-value)", colour = "trend",
       subtitle = "ratio = treatment/control") +
  xlim(c(-max(abs(pancreas_fc$log2_foldchange)),
         max(abs(pancreas_fc$log2_foldchange))))+
  scale_color_manual(values = c("grey"))+ 
  geom_text_repel(size = 2, max.overlaps = 20, show.legend = FALSE)

# top 10 (although all not significant)
arrange(pancreas_fc, -abs(log2_foldchange))[1:10,]

### CHST4: Carbohydrate Sulfotransferase - in Golgi,  modification of glycan structures on ligands of the lymphocyte homing receptor L-selectin
### (found an article where CHST15 was inhibited, which led to reduction of proliferation in PDAC)

### PABPC4: Poly(A) Binding Protein Cytoplasmic 4 


### ECI2:  Enoyl-CoA delta isomerase 2 - lipid metabolism 
### (found an article where ECI2 inhibition leads to down-regulated gene in cell cylce in prostate cancer)

```


### Comparing the two pathways on the basis of the cytokine dependent genes

After establishing the differences of the respective control and treatment data sets, the goal was to find genes that are related to pathways which either lead cells into uncontrolled proliferation or cell death. \
First, we extracted all 47 genes of the $\beta$-cell data set that showed a significant p-value in the Wilcoxon rank-sum test comparing treated and untreated cells. On the basis of a volcano plot we worked out specific genes that are significantly overexpressed in one data set, while showing little expression in the second one. The boxplot below shows the gene expression in healthy, stressed $\beta$- and pancreatic cancer cells. 

```{r}
#Selecting all the genes with a significant p-value for the Wilcoxon rank-sum test between the two diabetes data sets.

diabetes_fc_treatment = log2.fold.change(diabetes_GeneExprs_sub_treat, diabetes_GeneExprs_sub_control)
differentialgenesdiabetestreatment <- diabetes_fc_treatment[-which(diabetes_fc_treatment[,"wilcox_trend"]=="not significant"),1]
length(differentialgenesdiabetestreatment) 
# 47
differentialgenesdiabetestreatment

#Creating a matrix with only those 47 genes for diabetescontrol, diabetestreatment and pancreascontrol
diabetescontroldifferentialgenes <- diabetes_GeneExprs_sub_control[differentialgenesdiabetestreatment,]
diabetestreatmentdifferentialgenes <- diabetes_GeneExprs_sub_treat[differentialgenesdiabetestreatment,]
pancreascontroldifferentialgenes <- pancreas_GeneExprs_sub_control[differentialgenesdiabetestreatment,]

```

```{r}
# Creating the volcano Plot for the previously selected genes for diabetes Treatment vs. Pancreas Control
diabetesvspancreas_fc = log2.fold.change(pancreascontroldifferentialgenes, diabetestreatmentdifferentialgenes) # ratio = pancreas/diabetes

ggplot(data = diabetesvspancreas_fc, aes(x = log2_foldchange, y = wilcox_log10_pvalue, label = GeneSymbol, col = wilcox_trend)) + 
  geom_point() + 
  geom_hline(yintercept =  -1*log10(0.05), color = "red", linetype = "dashed") +
  labs(title = "Volcano plot (wilcox test) for PDAC control vs. beta-cell treatment ", 
       x = "log2 fold change", y = "-log10 (p-value)", colour = "trend",
       subtitle = "ratio = PDAC control / beta treatment (data sets: GSE59761 & GSE53454)") +
  xlim(c(-max(abs(diabetesvspancreas_fc$log2_foldchange)),
         max(abs(diabetesvspancreas_fc$log2_foldchange))))+ 
  scale_color_manual(values = c("navyblue", "gray50", "chartreuse4"),labels = c("lower in PDAC", "not significant", "higher in PDAC")) + 
  geom_text_repel(size = 2, max.overlaps = 25, show.legend = FALSE)

```

```{r}
# ------------------------------------------------
# boxplot: differentialgenesdiabetestreatment for the 3 data sets
# ------------------------------------------------

par(las = 2, 
    cex.axis = 0.5, cex.main = 0.8,
    mai = c(0.65,0.5,0.4,0.1), mfrow = c(2,1))

  
  boxplot(t(diabetes_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(diabetes_microarrays, treatment == "Cytok")$number]),
          col = rainbow(50),
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the treatment beta-cell data set"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
    boxplot(t(pancreas_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(pancreas_microarrays, siRNA == "NC")$number]),
          col = rainbow(50),
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the control PDAC data set"), 
          horizontal = FALSE)
  grid(NA,NULL)

  
  setwd(paste(projectPath, "plots", sep = "/"))
  # dev.copy2pdf(file = "boxplotcytokinegenesPDAC.pdf")  
  
par(las = 2, 
    cex.axis = 0.5, cex.main = 0.8,
    mai = c(0.65,0.5,0.4,0.1), mfrow = c(2,1))
  
  boxplot(t(diabetes_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(diabetes_microarrays, treatment == "Cytok")$number]),
          col = rainbow(50),
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the treatment beta-cell data set"), 
          horizontal = FALSE)
  grid(NA,NULL)

    boxplot(t(diabetes_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(diabetes_microarrays, treatment == "Control")$number]),
          col = rainbow(50), 
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the control beta-cell data set "), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  
  setwd(paste(projectPath, "plots", sep = "/"))
  # dev.copy2pdf(file = "boxplotcytokinegenes.pdf")



```

```{r fig.height=6}
# ------------------------------------------------
# boxplot: differentialgenesdiabetestreatment for the 3 data sets
# ------------------------------------------------

par(las = 2, 
    cex.axis = 0.9, cex.main = 1.3,
    mai = c(0.7,0.5,0.4,0.1), mfrow = c(3,1))

  
  boxplot(t(diabetes_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(diabetes_microarrays, treatment == "Cytok")$number]),
          col = rainbow(50),
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the treatment beta-cell data set"), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  boxplot(t(diabetes_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(diabetes_microarrays, treatment == "Control")$number]),
          col = rainbow(50), 
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the control beta-cell data set "), 
          horizontal = FALSE)
  grid(NA,NULL)
  
  
    boxplot(t(pancreas_GeneExprs_sub_combined[differentialgenesdiabetestreatment,filter(pancreas_microarrays, siRNA == "NC")$number]),
          col = rainbow(50),
          main= paste0("Boxplot of cytokine dependent TRA gene expression in the control PDAC data set"), 
          horizontal = FALSE)
  grid(NA,NULL)

```

```{r fig.cap = "The boxplots depicts the 47 cytokine-dependent genes and their differing gene expression in untreated (red), treated beta- (green) and pancreatic cancer cells (blue)", include=TRUE, eval=TRUE, out.width="100%", fig.width=15, fig.height=6}
# Combined Boxplot of the cytokine dependent genes of the 3 data sets

boxplotmatrix <- cbind(diabetescontroldifferentialgenes,diabetestreatmentdifferentialgenes,pancreascontroldifferentialgenes)
mel2 <- melt(t(boxplotmatrix))

# vector to annotate the values from mel with treatment

treatm <- diabetes_microarrays[,"treatment"] 
names(treatm) <- diabetes_microarrays[,"number"]
treatm = unname(treatm[as.character(mel2$Var1)])
#Replacing Na with Pancreas cancer
treatm <- replace_na(treatm, "Pancreas")


# create data frame (used for ggplot)
boxplotmatrix2 <- cbind(treatment = treatm , mel2)

theme_set(theme_bw())
ggplot(boxplotmatrix2, aes(x = Var2, y = value, color = treatment)) + 
  geom_boxplot(position = position_dodge(1), lwd = 1.1) + 
  labs(title = "Different gene expression of data sets in comparison",
       color = "cell type",
       x = "genes", y = "gene expression") + 
  theme(axis.text.x = element_text(angle = 90, size = 17),
        axis.text.y = element_text(size = 17),
        axis.title = element_text(size = 20),
        legend.title = element_text(size = 25),
        legend.text = element_text(size = 20),
        plot.title = element_text(size = 25)) +
  scale_color_manual(values = c("firebrick1", "green3", "cornflowerblue"),
                    labels = c("Control", "Cytokine", "PDAC")) 
  

```

The two genes that are the highest expressed genes in pancreatic cancer and lowly expressed in diabetes are GMNN and STT3A. \
Geminin (GMNN) was previously discussed and plays an important role in the regulation of DNA replication. \
STT3A is upregulated in pancreatic cancer compared to the expression in the normal pancreas (Pan \textit{et al.}, 2014). STT3A is a subunit of oligosaccharyltransferase (OST), which is a membrane protein complex catalyzing Asn-linked glycosylation of polypeptides (Ruiz-Canada \textit{et al.}, 2009). Epithelial-to-mesenchymal transition (EMT) in tumor cells induces upregulation of OST to ensure cell surface expression of PD-L1, leading to immune evasion (Harada \textit{et al.}, 2019). \
The high expression of those two genes is therefore being proven to be part of the pathways that lead cells into tumorigenesis. The related pathways are regulation of DNA replication and OST. They result in genomic instability as well as immune evasion. Hence, overexpression of GMNN and STT3A promotes tumor growth. \
Meanwhile, finding genes that are highly expressed in diabetes and lowly expressed in pancreatic cancer was challenging. Some of the found genes related to pathways introducing $\beta$-cell death and therefore diabetes, were ambiguous. Two examples were chosen to demonstrate difficulties that were encountered as well as one gene which reflects the desired results. \
The highest differentially expressed gene in diabetes is SPINK1. Serine protease inhibitor Kazal type 1 (SPINK1) is a trypsin inhibitor maintaining the integrity of pancreatic tissue, which is associated with pancreatitis as well as diabetes (Muller \textit{et al.}, 2019). There is little information up to this date solely concerning the role of SPINK1 in diabetes. Most published scientific papers focus on the role of SPINK1 in relation to pancreatitis and how this disease correlates with diabetes and pancreatic cancer. \
Another example is REG1A, which encodes a protein called Pancreatic Stone Protein (PSP). A high expression and subsequent secretion of PSP is induced by $\beta$-cells undergoing apoptosis in neighboring pancreatic cells (Bacon \textit{et al.}, 2012). PSP is associated with islet regeneration and proliferation as a reaction to the $\beta$-cell loss in diabetes (Yang \textit{et al.}, 2015). However, REG1A is also linked to pancreatic cancer. A high expression of REG1A is associated with a poor prognosis in pancreatic cancer, especially in patients with diabetes, as it promotes cell proliferation and tumor growth (Zhou \textit{et al.}, 2010). Evidently, that was not the case in our data set, but since the gene is associated with both diabetes and cancer, it cannot be used to predict the fate of pancreatic cells. \
SPINK1 and REG1A can not be applicated in the prediction of diabetes, because of insufficient research of the respective genes as well as a strong relation to pancreatic cancer. Nevertheless, we also found genes that are only associated with diabetes. \
A successfully identified gene is CXCL11, which was previously mentioned as upregulated in cells treated with cytokines. CXCL11 is a chemokine that regulates immune cell migration and induces inflammation (Kochumon \textit{et al.}, 2020). Diabetes is induced by the ensuing $\beta$-cell death (Burke \textit{et al.}, 2016). \
There are quite a few genes that are expressed differently in pancreatic cancer and diabetes compared to normal $\beta$-cells. However, the most prominent ones are GMNN and STT3A for a high cancer expression as well as SPINK1 and REG1A for diabetes. They induce aforementioned pathways leading $\beta$-cells to their respective fates. 

## Linear regression model

A linear regression model was applied to emulate the influence of time on a gene in untreated $\beta$-cells. \
To identify highly time-influenced genes we first extracted genes, which had a high influence on PC1 (known to be impacted by time) and a low influence on PC2. Then we looked at the remaining genes in a heatmap and chose the gene with the most variation: CUZD1. It activates trypsinogen, the precursor of the pancreas enzyme trypsin, and may regulate cell motility, cell-cell and cell-ECM interactions (Leong et. al., 2004). 

```{r}
# ------------------------------------------------
# which genes are influenced by time 
# ------------------------------------------------

find_timegenes <- function(t1 = 0.1, t2 = 0.05){
  g1 = sort(abs(diabetes_pca$rotation[,"PC1"]), decreasing = TRUE)
  g2 = sort(abs(diabetes_pca$rotation[,"PC2"]))
  
  n1 = names(g1[which(g1 > t1)])
  n2 = names(g2[which(g2 < t2)])
  
  list(PC1_high = n1, PC2_low = n2, PC1_high_PC2_low = which(n2 %in% n1))
}



# genes with high PC1
n1 = find_timegenes(0.1, 0.05)$PC1_high
# genes with PC1 high and PC2 low
n2 = find_timegenes(0.1, 0.05)$PC2_low[find_timegenes(0.1, 0.05)$PC1_high_PC2_low]


pheatmap(diabetes_GeneExprs_sub_combined[n2, arrange(filter(diabetes_microarrays, treatment == "Cytok"), time_h)$number], 
         main = "gene expression over time for cytokine treated ß-cells",
         cluster_rows = FALSE, cluster_cols = FALSE)
```


```{r fig.cap = "The heatmap shows all genes in untreated beta-cells that are highly influenced by time. CUZD1 has the most variation.", include=TRUE, eval=TRUE, fig.width=9, fig.height=5}

anno <- select(filter(diabetes_microarrays, treatment == "Control"), time_h)
rownames(anno) <- filter(diabetes_microarrays, treatment == "Control")$number

pheatmap(diabetes_GeneExprs_sub_combined[n2, arrange(filter(diabetes_microarrays, treatment == "Control"), time_h)$number], 
         main = "gene expression over time for beta-cells",
         cluster_rows = FALSE, 
         cluster_cols = FALSE,
         annotation_col = anno,
         angle_col = 315,
         border_color = NA,
         fontsize = 15) 

# remove(n1,n2)
# suggestion for linear regression: CUZD1 (Control), PNLIP (Cytokine treatment)
```

```{r}
# ------------------------------------------------
# compare linear regression and mean model
# ------------------------------------------------
CUZD1exprdieabcontr <- matrix(
  c(0,1,2,4,8,12,24,36,48,60,72,84,diabetes_GeneExprs_sub_control[56,1],
 diabetes_GeneExprs_sub_control[56,3],
diabetes_GeneExprs_sub_control[56,5],
diabetes_GeneExprs_sub_control[56,8],
 diabetes_GeneExprs_sub_control[56,12],
diabetes_GeneExprs_sub_control[56,2],
 diabetes_GeneExprs_sub_control[56,4],
diabetes_GeneExprs_sub_control[56,6],
 diabetes_GeneExprs_sub_control[56,7],
diabetes_GeneExprs_sub_control[56,9],
 diabetes_GeneExprs_sub_control[56,10],
diabetes_GeneExprs_sub_control[56,11]),
ncol=2)

colnames(CUZD1exprdieabcontr) <- c("time", "expression")
rownames(CUZD1exprdieabcontr) <-c("GSM1293805",
 "GSM1293807","GSM1293809", "GSM1293812", "GSM1293816","GSM1293806", "GSM1293808","GSM1293810", "GSM1293811","GSM1293813", "GSM1293814", "GSM1293815")

b1 = cor(CUZD1exprdieabcontr[,1], CUZD1exprdieabcontr[,2]) * (sd(CUZD1exprdieabcontr[,2])/sd(CUZD1exprdieabcontr[,1]))
b0 = mean(CUZD1exprdieabcontr[,2]) - b1*mean(CUZD1exprdieabcontr[,1])
modelvalue = b0 +b1*96
modelprediction <- matrix(c(96, modelvalue), nrow = 1)
colnames(modelprediction) <- c("time", "expression")
rownames(modelprediction) <- c("modelprediciton")

modelvalues <- c(b0 +b1*0, b0 +b1* 1 ,b0 +b1*2, b0 +b1*4, b0 +b1*8, b0 +b1*12, b0 +b1*24, b0 +b1*36, b0 +b1*48, b0 +b1*60, b0 +b1*72, b0 +b1*84, b0 + b1*96)
remove(b0,b1)
modelvaluesmatrix <- matrix(c(0,1,2,4,8,12,24,36,48,60,72,84,96, modelvalues), ncol=2)
colnames(modelvaluesmatrix) <- c("time", "expression")

meanprediction <- matrix(c(96, mean(CUZD1exprdieabcontr[,2])), nrow=1)
colnames(meanprediction) <- c("time", "expression")
rownames(meanprediction) <- c("meanprediciton")

realvalue <- matrix(c(96,diabetes_GeneExprs_sub_control[56,13]), nrow=1)
colnames(realvalue) <- c("time", "expression")
rownames(realvalue) <- c("GSM1293817")
```
Using only the time points from 0h to 84h, we developed a linear regression model to predict the expression of CUZD1 at 96h and compared this value to the actual value as well as to a value predicted by a mean-based model, shown in the plot below.

```{r fig.cap = "The expression of CUZD1 at different time points (grey) is used to develop a linear regression model (red) to predict the value at 96h. The mean-based model is depicted in blue.", include=TRUE, eval=TRUE}
plot(CUZD1exprdieabcontr, pch=20,col='grey', main="Models of CUZD1 expression \nin untreated ß-cells", xlim=c(0, 100), ylim=c(8, 13))
points(modelprediction, col="red", pch=20)
points(meanprediction, col="blue", pch=20)
points(realvalue, col="black", pch=20)

lines(c(0,1,2,4,8,12,24,36,48,60,72,84,96),modelvalues, col="indianred1", lwd=2)

lines(0:96, rep(mean(CUZD1exprdieabcontr[,2]), 97), col="lightblue", lwd=2)


legend("bottomleft", 
  legend = c("mean", "linear model", "true value"), 
  col = c("blue", "red", "black"), 
  pch = c(19,19, 19), 
  bty = "n", 
  pt.cex = 1, 
  cex = 0.65, 
  text.col = "black", 
  horiz = F , 
  inset = c(0.05, 0.1))
```
A F-test confirmed that the variable time is very significant
$(p<10^{-4})$. Also, over 80% of the overall variance is explained by the model. Finally, the linear model as a whole is superior to the mean model. \

```{r}
## F-test

CUZD1exprdieabcontr.df <- data.frame(
   time = c(0,1,2,4,8,12,24,36,48,60,72,84), 
   expression = c(diabetes_GeneExprs_sub_control[56,1],
 diabetes_GeneExprs_sub_control[56,3],
diabetes_GeneExprs_sub_control[56,5],
diabetes_GeneExprs_sub_control[56,8],
 diabetes_GeneExprs_sub_control[56,12],
diabetes_GeneExprs_sub_control[56,2],
 diabetes_GeneExprs_sub_control[56,4],
diabetes_GeneExprs_sub_control[56,6],
 diabetes_GeneExprs_sub_control[56,7],
diabetes_GeneExprs_sub_control[56,9],
 diabetes_GeneExprs_sub_control[56,10],
diabetes_GeneExprs_sub_control[56,11]),
   stringsAsFactors = FALSE)
rownames(CUZD1exprdieabcontr) <-c("GSM1293805",
 "GSM1293807","GSM1293809", "GSM1293812", "GSM1293816","GSM1293806", "GSM1293808","GSM1293810", "GSM1293811","GSM1293813", "GSM1293814", "GSM1293815")

CUZD1diabctr.lm = lm(expression ~ time, data=CUZD1exprdieabcontr.df)

summary(CUZD1diabctr.lm) 

```

# Discussion


\newpage
## References

Bacon, S., Kyithar, M. P., Schmid, J., Rizvi, S. R., Bonner, C., Graf, R., Prehn, J. H., & Byrne, M. M. (2012). Serum levels of pancreatic stone protein (PSP)/reg1A as an indicator of beta-cell apoptosis suggest an increased apoptosis rate in hepatocyte nuclear factor 1 alpha (HNF1A-MODY) carriers from the third decade of life onward. BMC Endocr. Disord. \textit{12}, 13. \

Burke, S. J., Karlstad, M. D., Eder, A. E., Regal, K. M., Lu, D., Burk, D. H., & Collier, J. J. (2016). Pancreatic $\beta$-Cell production of CXCR3 ligands precedes diabetes onset. BioFactors \textit{42}, 703–715. \

Eizirik, D. L., Pasquali, L., & Cnop, M. (2020). Pancreatic $\beta$-cells in type 1 and type 2 diabetes mellitus: different pathways to failure. Nat. Rev. Endocrinol. \textit{16}, 349–362. \

Harada, Y., Ohkawa, Y., Kizuka, Y., & Taniguchi, N. (2019). Oligosaccharyltransferase: A Gatekeeper of Health and Tumor Progression. Int. J. Mol. Sci. \textit{20}, 6074. \

Kochumon, S., Madhoun, A. A., Al-Rashed, F., Azim, R., Al-Ozairi, E., Al-Mulla, F., & Ahmad, R. (2020). Adipose tissue gene expression of CXCL10 and CXCL11 modulates inflammatory markers in obesity: implications for metabolic inflammation and insulin resistance. Ther. Adv. Endocrinol. Metab. \textit{11}, 2042018820930902. \

Kushwaha, P. P., Rapalli, K. C., & Kumar, S. (2016). Geminin a multi task protein involved in cancer pathophysiology and developmental process: A review. Biochimie \textit{131}, 115–127. \

Muller, N., Sarantitis I., Rouanet, M., de Mestier, L., Halloran, C., Greenhalf, W., Férec, C., Masson, E., Ruszniewski, P., Lévy, P., Neoptolemos, J., Buscail, L. & Rebours V. (2019). Natural history of SPINK1 germline mutation related-pancreatitis. EBioMedicine \textit{48}, 581–591. \

Pan, S., Chen, R., Tamura, Y., Crispin, D. A., Lai, L. A., May, D. H., McIntosh, M. W., Goodlett, D. R., & Brentnall, T. A. (2014). Quantitative glycoproteomics analysis reveals changes in N-glycosylation level associated with pancreatic ductal adenocarcinoma. J. Proteome Res. \textit{13}, 1293–1306. \

Ruiz-Canada, C., Kelleher, D. J., & Gilmore, R. (2009). Cotranslational and posttranslational N-glycosylation of polypeptides by distinct mammalian OST isoforms. Cell \textit{136}, 272–283. \

Salabat, M. R., Melstrom, L. G., Strouch, M. J., Ding, X. Z., Milam, B. M., Ujiki, M. B., Chen, C., Pelling, J. C., Rao, S., Grippo, P. J., McGarry, T. J., & Bentrem, D. J. (2008). Geminin is overexpressed in human pancreatic cancer and downregulated by the bioflavanoid apigenin in pancreatic cancer cell lines. Mol. Carcinog. \textit{47}, 835–844. \

Takakura, K., Shibazaki, Y., Yoneyama, H., Fujii, M., Hashiguchi, T., Ito, Z., Kajihara, M., Misawa, T., Homma, S., Ohkusa, T., & Koido, S. (2015). Inhibition of Cell Proliferation and Growth of Pancreatic Cancer by Silencing of Carbohydrate Sulfotransferase 15 In Vitro and in a Xenograft Model. PloS one \textit{10}, e0142981. \

Tokunaga, R., Zhang, W., Naseem, M., Puccini, A., Berger, M. D., Soni, S., McSkane, M., Baba, H., & Lenz, H. J. (2018). CXCL9, CXCL10, CXCL11/CXCR3 axis for immune activation - A target for novel cancer therapy. Cancer Treat Rev. \textit{63}, 40–47. \

Yang, J., Li, L., Raptis, D., Li, X., Li, F., Chen, B., He, J., Graf, R., & Sun, Z. (2015). Pancreatic stone protein/regenerating protein (PSP/reg): a novel secreted protein up-regulated in type 2 diabetes mellitus. Endocrine \textit{48}, 856–862. \

Zhou, L., Zhang, R., Wang, L., Shen, S., Okamoto, H., Sugawara, A., Xia, L., Wang, X., Noguchi, N., Yoshikawa, T., Uruno, A., Yao, W., & Yuan, Y. (2010). Upregulation of REG Ialpha accelerates tumor progression in pancreatic cancer with diabetes. Int. J. Cancer. \textit{127}, 1795–1803. 

Quellen Biomarker:

Kim, H., Kang, K. N., Shin, Y. S., Byun, Y., Han, Y., Kwon, W., Kim, C. W., & Jang, J. Y. (2020). Biomarker Panel for the Diagnosis of Pancreatic Ductal Adenocarcinoma. Cancers, 12(6), 1443.


McGuigan A, Kelly P, Turkington RC, Jones C, Coleman HG, McCain RS. Pancreatic cancer: A review of clinical diagnosis, epidemiology, treatment and outcomes. World J Gastroenterol. 2018 Nov 21;24(43):4846-4861. doi: 10.3748/wjg.v24.i43.4846. PMID: 30487695; PMCID: PMC6250924.

Heo K, Lee S. TSPAN8 as a Novel Emerging Therapeutic Target in Cancer for Monoclonal Antibody Therapy. Biomolecules. 2020 Mar 3;10(3):388. doi: 10.3390/biom10030388. PMID: 32138170; PMCID: PMC7175299.

Wang H, Rana S, Giese N, Büchler MW, Zöller M. Tspan8, CD44v6 and alpha6beta4 are biomarkers of migrating pancreatic cancer-initiating cells. Int J Cancer. 2013 Jul 15;133(2):416-26. doi: 10.1002/ijc.28044. Epub 2013 Feb 15. PMID: 23338841.


